<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.313">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kyle Palos">

<title>Figure4</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="figure4_files/libs/clipboard/clipboard.min.js"></script>
<script src="figure4_files/libs/quarto-html/quarto.js"></script>
<script src="figure4_files/libs/quarto-html/popper.min.js"></script>
<script src="figure4_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="figure4_files/libs/quarto-html/anchor.min.js"></script>
<link href="figure4_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="figure4_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="figure4_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="figure4_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="figure4_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Figure4</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Kyle Palos </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<p>This script performs the analyses going into Figure 4 and associated supplement of the comparative PRM manuscript</p>
<p>Most of these analyses are from the concatenated ModTect output from this analysis of mRNA degradation in Arabidopsis:</p>
<p><a href="https://www.pnas.org/doi/10.1073/pnas.1712312115" class="uri">https://www.pnas.org/doi/10.1073/pnas.1712312115</a></p>
<p>Data is hosted on FigShare</p>
<section id="download-mods-and-metadata" class="level2">
<h2 class="anchored" data-anchor-id="download-mods-and-metadata">Download mods and metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/45555360"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="at">destfile =</span> <span class="st">"~/Downloads/at_degradation.txt.gz"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/degradation_experiment_metadata.txt?raw=TRUE"</span>,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">"~/Downloads/at_degradation_metadata.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="read-in-files" class="level3">
<h3 class="anchored" data-anchor-id="read-in-files">read in files</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>deg <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/at_degradation.txt.gz"</span>, </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">col_names =</span> T, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(file,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"SRA"</span>, <span class="st">"other"</span>),</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"_GSM"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 552926 Columns: 45
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (11): chrom, reference_nt, gene_chr, gene, code, strand, source, type, o...
dbl (34): position, position2, mono-alleleLogP, bi-alleleLogP, tri-alleleLog...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/at_degradation_metadata.txt"</span>, </span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 128 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): Run, Experiment

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>deg <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., samples, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"SRA"</span> <span class="ot">=</span> <span class="st">"Run"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Keep PRMs present in at least 2 replicates:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>sites_to_keep <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(full_position) <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_experiments =</span> <span class="fu">n_distinct</span>(SRA)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_experiments <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>deg <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., sites_to_keep, <span class="at">by =</span> <span class="st">"full_position"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>deg<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gene:"</span>, <span class="st">""</span>, deg<span class="sc">$</span>gene)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># load in accessory data files</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>decay <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Desktop/Nelson_lab/epitranscriptomics/mini_comparative_project/arabidopsis_degradation/large_quants/decay_model.txt"</span>, </span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>                          <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>                          <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18674 Columns: 31
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (3): gene, alphaPattern, betaPattern
dbl (28): alpha_WT, alpha_sov, alpha_vcs, alpha_vcs.sov, beta_WT, beta_sov, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>rpkm <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Desktop/Nelson_lab/epitranscriptomics/mini_comparative_project/arabidopsis_degradation/large_quants/rpkm.txt"</span>, </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18674 Columns: 37
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (1): geneID
dbl (36): RPKM_VS_00, RPKM_Vs_00, RPKM_vS_00, RPKM_vs_00, VS_00_mean, VS_07_...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>deg_results <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Desktop/Nelson_lab/epitranscriptomics/mini_comparative_project/arabidopsis_degradation/large_quants/deg_results.txt"</span>, </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18674 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (1): geneID
dbl (13): baseMean, log2FC_Vs-VS, log2FC_vs-Vs, log2FC_vs-VS, log2FC_vs-vS, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>load in supplemental data. There was some excel formatting for this supplemental data that I cleaned, which is why I’m providing the links to my github and not directly to the paper’s data.</p>
<p>Using the authors’ computed RPKM values for RNA abundance as they have accounted for degradation specific</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>rpkm <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/rpkm_clean.txt?raw=TRUE"</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(rpkm, <span class="st">"~/Downloads/rpkm.txt"</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>decay <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/decay_model.txt?raw=TRUE"</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(decay, <span class="st">"~/Downloads/decay.txt"</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co"># differentially expressed genes</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>diffexp <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/deg_results.txt?raw=TRUE"</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(diffexp, <span class="st">"~/Downloads/deg.txt"</span>)</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>rpkm <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/rpkm.txt"</span>, <span class="at">col_names =</span> T, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18674 Columns: 33
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (1): gene
dbl (32): WT_00_mean, WT_07_mean, WT_15_mean, WT_30_mean, WT_60_mean, WT_120...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>decay <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/decay.txt"</span>, <span class="at">col_names =</span> T, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18674 Columns: 31
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (3): gene, alphaPattern, betaPattern
dbl (28): alpha_WT, alpha_sov, alpha_vcs, alpha_vcs.sov, beta_WT, beta_sov, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>diffexp <span class="ot">&lt;-</span>  <span class="fu">read_delim</span>(<span class="st">"~/Downloads/deg.txt"</span>, <span class="at">col_names =</span> T, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18674 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr  (1): geneID
dbl (13): baseMean, log2FC_Vs-VS, log2FC_vs-Vs, log2FC_vs-VS, log2FC_vs-vS, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>load in the number of mapped reads per SRA:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/arabidopsis_large_deg_mapped_reads.txt?raw=TRUE"</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="at">destfile =</span> <span class="st">"~/Downloads/mapped.txt"</span>)</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>mapped <span class="ot">&lt;-</span> <span class="fu">read_table</span>(<span class="st">"~/Downloads/mapped.txt"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">SRA =</span> <span class="dv">1</span>,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">mapped =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  SampleID = col_character(),
  TotalMappedReads = col_double()
)</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>deg <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Experiment, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"time"</span>), <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> F) </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>deg<span class="sc">$</span>time <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(deg<span class="sc">$</span>time)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Establish the decay groups between mRNAs modified at TP0 and those not modified at TP0. Then plot the alpha decay rates of those groups</p>
</section>
</section>
<section id="figure-4e-supp-figure-9d" class="level1">
<h1>Figure 4E &amp; Supp Figure 9D</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># these are the genes modified at time point 0</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>genes_mod_time_0 <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co"># evaluate the decay rate of RNAs that had HAMR PTMs</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>decay_mod <span class="ot">&lt;-</span> decay <span class="sc">%&gt;%</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., genes_mod_time_0, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">"modified"</span>)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>decay_not_mod <span class="ot">&lt;-</span> decay <span class="sc">%&gt;%</span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(., genes_mod_time_0, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">"not modified"</span>)</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>decay <span class="ot">&lt;-</span> <span class="fu">rbind</span>(decay_mod, decay_not_mod)  </span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>decay <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> type, <span class="at">y =</span> alpha_WT, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_violin(trim = F) +</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  coord_cartesian(ylim = c(1e-08,9e-02)) +</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>)) <span class="sc">+</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"α decay rate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in scale_y_log10(): log-10 transformation introduced infinite values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 89 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Supp 9D</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>decay <span class="sc">%&gt;%</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> type, <span class="at">y =</span> alpha_WT, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  coord_cartesian(ylim = c(1e-08,9e-02)) +</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>)) <span class="sc">+</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"α decay rate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in scale_y_log10(): log-10 transformation introduced infinite values.
Removed 89 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-7-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 4E</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>decay <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> type, <span class="at">y =</span> alpha_WT, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(.<span class="dv">0001</span>, .<span class="dv">001</span>, .<span class="dv">01</span>, .<span class="dv">1</span>), <span class="at">label =</span> comma) <span class="sc">+</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#coord_flip() +</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a> <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">1e-04</span>,<span class="fl">1e-01</span>)) <span class="sc">+</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>)) <span class="sc">+</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"α decay rate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in scale_y_log10(breaks = c(1e-04, 0.001, 0.01, 0.1), label = comma): log-10 transformation introduced infinite values.
Removed 89 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-7-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(decay_mod<span class="sc">$</span>alpha_WT,</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>       decay_not_mod<span class="sc">$</span>alpha_WT)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  decay_mod$alpha_WT and decay_not_mod$alpha_WT
t = -16.37, df = 628.74, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.006156528 -0.004837658
sample estimates:
  mean of x   mean of y 
0.004170579 0.009667673 </code></pre>
</div>
</div>
<p>Now plot the alpha groups that modified and not-modified mRNAs belong to:</p>
</section>
<section id="figure-4f" class="level1">
<h1>Figure 4F</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>decay <span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">group2 =</span> <span class="fu">case_when</span>(alpha_grp <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">&amp;</span> alpha_grp <span class="sc">&lt;</span> <span class="dv">15</span> <span class="sc">~</span> <span class="st">"genotype dependent"</span>,</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>                           alpha_grp <span class="sc">==</span> <span class="dv">15</span> <span class="sc">~</span> <span class="st">"genotype independent (exosome)"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> alpha_grp, <span class="at">fill =</span> <span class="fu">str_wrap</span>(group2, <span class="dv">15</span>))) <span class="sc">+</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">"white"</span>, <span class="at">binwidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>type, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"lightgrey"</span>, <span class="st">"black"</span>),</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"α group"</span>) <span class="sc">+</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of RNAs"</span>) <span class="sc">+</span></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">15</span>),</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># associated chi-squared test:</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>for_chi <span class="ot">&lt;-</span> decay <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(gene, alpha_grp, type) <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(alpha_grp, type) <span class="sc">%&gt;%</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n_distinct</span>(gene)) <span class="sc">%&gt;%</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> type, <span class="at">values_from =</span> n) <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"alpha_grp"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'alpha_grp'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>chisq <span class="ot">&lt;-</span> <span class="fu">chisq.test</span>(for_chi)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>chisq<span class="sc">$</span>observed</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   modified not modified
1        10          934
2        13          482
3        16          821
4        39         1104
5        73         2096
6        19         1094
7         6          373
8        17          800
9        12          556
10       18         1155
11       74         3126
12       80         3091
13       28          572
14       23          673
15      117         1252</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>chisq<span class="sc">$</span>expected</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   modified not modified
1  27.55061     916.4494
2  14.44656     480.5534
3  24.42781     812.5722
4  33.35841    1109.6416
5  63.30218    2105.6978
6  32.48286    1080.5171
7  11.06110     367.9389
8  23.84411     793.1559
9  16.57706     551.4229
10 34.23396    1138.7660
11 93.39188    3106.6081
12 92.54552    3078.4545
13 17.51098     582.4890
14 20.31273     675.6873
15 39.95421    1329.0458</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>chisq<span class="sc">$</span>p.value</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.814756e-35</code></pre>
</div>
</div>
<p>Next, we’re going to compare the types of RNA being modified at the 0 time point and 8 hours after arresting transcription.</p>
<p>Load in Araport11 gene types, you can get this data here:</p>
<p><a href="https://www.arabidopsis.org/download_files/Genes/Araport11_genome_release/Araport11_gene_type" class="uri">https://www.arabidopsis.org/download_files/Genes/Araport11_genome_release/Araport11_gene_type</a></p>
<p>Or my Github:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/araport11_gene_types.txt?raw=TRUE"</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="at">destfile =</span> <span class="st">"~/Downloads/types.txt"</span>)</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>gene_types <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/types.txt"</span>, <span class="at">header =</span> T)</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>gene_types <span class="ot">&lt;-</span> gene_types <span class="sc">%&gt;%</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">gene =</span> <span class="dv">1</span>, <span class="at">type =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-4a" class="level1">
<h1>Figure 4A:</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>deg <span class="sc">%&gt;%</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(full_position, gene, Experiment) <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., gene_types, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Experiment, type) <span class="sc">%&gt;%</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_mod_genes =</span> <span class="fu">n_distinct</span>(gene)) <span class="sc">%&gt;%</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Experiment,</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"time"</span>),</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> time <span class="sc">==</span> <span class="dv">480</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type2 =</span> <span class="fu">case_when</span>(type <span class="sc">==</span> <span class="st">"small_nuclear_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"small_nucleolar_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"antisense_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"antisense_long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"other_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"ribosomal_rna"</span> <span class="sc">~</span> <span class="st">"rRNA"</span>,</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"protein_coding"</span> <span class="sc">~</span> <span class="st">"mRNA"</span>,</span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"pre_trna"</span> <span class="sc">~</span> <span class="st">"tRNA"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type2 <span class="sc">!=</span> <span class="st">"miRNA_primary_transcript"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">time2 =</span> <span class="fu">case_when</span>(time <span class="sc">==</span> <span class="st">"0"</span> <span class="sc">~</span> <span class="st">"Time point: 0"</span>,</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>                           time <span class="sc">==</span> <span class="st">"480"</span> <span class="sc">~</span> <span class="st">"Time point: 8hr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(type2, time2) <span class="sc">%&gt;%</span></span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_genes =</span> <span class="fu">sum</span>(n_mod_genes)) <span class="sc">%&gt;%</span></span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> type2, <span class="at">y =</span> n_genes, <span class="at">fill =</span> type2)) <span class="sc">+</span></span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> n_genes),<span class="at">size =</span> <span class="dv">16</span>, <span class="at">hjust =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>time2) <span class="sc">+</span></span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"number of modified transcripts"</span>) <span class="sc">+</span></span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Experiment'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'type2'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>load in Salmon/tximport TPM values. This was done before utilizing the RPKM values to simply determine which genes were expressed (detectable) or not.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/arabidopsis_degradation_tpm.txt.gz?raw=TRUE"</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/at_degradation_tpm.txt.gz"</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>tpm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/at_degradation_tpm.txt.gz"</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The following code generates:</p>
</section>
<section id="supplemental-figure-9b" class="level1">
<h1>Supplemental Figure 9B</h1>
</section>
<section id="figure-4b" class="level1">
<h1>Figure 4B</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize the mean TPM per experiment (genotype by time-point)</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>tpm_sum <span class="ot">&lt;-</span> tpm <span class="sc">%&gt;%</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># left_join(., types, by = "gene") %&gt;%</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., samples, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"name"</span> <span class="ot">=</span> <span class="st">"Run"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., gene_types, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Experiment, gene, type) <span class="sc">%&gt;%</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_tpm =</span> <span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Experiment', 'gene'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the number of expressed genes per genotype/time-point</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>tpm_sum2 <span class="ot">&lt;-</span> tpm_sum <span class="sc">%&gt;%</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Experiment, type) <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_expressed_genes =</span> <span class="fu">sum</span>(mean_tpm <span class="sc">&gt;=</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Experiment'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make the genotype and time components in separate columns</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="ot">&lt;-</span> tpm_sum2 <span class="sc">%&gt;%</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Experiment,</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"time"</span>),</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a><span class="co"># make the genotype a factor and set the level</span></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>summary_combined<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">factor</span>(summary_combined<span class="sc">$</span>genotype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"WT"</span>, <span class="st">"sov"</span>, <span class="st">"varicose"</span>, <span class="st">"double"</span>))</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a><span class="co"># re-code the gene types</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="ot">&lt;-</span> summary_combined <span class="sc">%&gt;%</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type2 =</span> <span class="fu">case_when</span>(type <span class="sc">==</span> <span class="st">"small_nuclear_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"small_nucleolar_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"antisense_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"antisense_long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"other_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"ribosomal_rna"</span> <span class="sc">~</span> <span class="st">"rRNA"</span>,</span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"protein_coding"</span> <span class="sc">~</span> <span class="st">"PCG"</span>,</span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>                         type <span class="sc">==</span> <span class="st">"pre_trna"</span> <span class="sc">~</span> <span class="st">"tRNA"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type2 <span class="sc">!=</span> <span class="st">"miRNA_primary_transcript"</span>) </span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a><span class="co"># supplemental figure 9b</span></span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="sc">%&gt;%</span></span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(genotype, time, type2) <span class="sc">%&gt;%</span></span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_expr =</span> <span class="fu">sum</span>(n_expressed_genes)) <span class="sc">%&gt;%</span></span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-30"><a href="#cb56-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type2 <span class="sc">!=</span> <span class="st">"rRNA"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-31"><a href="#cb56-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(time), <span class="at">y =</span> n_expr, <span class="at">color =</span> genotype)) <span class="sc">+</span></span>
<span id="cb56-32"><a href="#cb56-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.85</span>)<span class="sc">+</span></span>
<span id="cb56-33"><a href="#cb56-33" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb56-34"><a href="#cb56-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># facet_wrap(genotype~name, scales = "free") +</span></span>
<span id="cb56-35"><a href="#cb56-35" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ggh4x::facet_grid2(genotype~name, scales = "free_y", independent = "y") +</span></span>
<span id="cb56-36"><a href="#cb56-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb56-37"><a href="#cb56-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>)) <span class="sc">+</span></span>
<span id="cb56-38"><a href="#cb56-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>type2, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb56-39"><a href="#cb56-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"time (minutes)"</span>) <span class="sc">+</span></span>
<span id="cb56-40"><a href="#cb56-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of expressed genes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'genotype', 'time'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summarize the number of PRMs per experiment for plotting</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>mod_summary <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Experiment) <span class="sc">%&gt;%</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_genes =</span> <span class="fu">n_distinct</span>(gene),</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_sites =</span> <span class="fu">n_distinct</span>(full_position)) <span class="sc">%&gt;%</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="ot">&lt;-</span> tpm_sum2 <span class="sc">%&gt;%</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., mod_summary, <span class="at">by =</span> <span class="st">"Experiment"</span>)</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="ot">&lt;-</span> summary_combined <span class="sc">%&gt;%</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Experiment,</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"time"</span>),</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"n expr genes"</span> <span class="ot">=</span> <span class="dv">4</span>,</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>                <span class="st">"n modified RNAs"</span> <span class="ot">=</span> <span class="dv">5</span>,</span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>                <span class="st">"n modified sites"</span> <span class="ot">=</span> <span class="dv">6</span>)</span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>summary_combined<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">factor</span>(summary_combined<span class="sc">$</span>genotype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"WT"</span>, <span class="st">"sov"</span>, <span class="st">"varicose"</span>, <span class="st">"double"</span>))</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Figure 4B </span></span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="sc">%&gt;%</span></span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_norm =</span> <span class="st">`</span><span class="at">n modified RNAs</span><span class="st">`</span><span class="sc">/</span><span class="st">`</span><span class="at">n expr genes</span><span class="st">`</span>,</span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">density =</span> <span class="st">`</span><span class="at">n modified sites</span><span class="st">`</span><span class="sc">/</span><span class="st">`</span><span class="at">n modified RNAs</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"n modified normalized"</span> <span class="ot">=</span> gene_norm,</span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">mod_density =</span> density) <span class="sc">%&gt;%</span></span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(genotype,time, type)) <span class="sc">%&gt;%</span></span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">==</span> <span class="st">"n modified sites"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(time), <span class="at">y =</span> value, <span class="at">color =</span> genotype)) <span class="sc">+</span></span>
<span id="cb58-32"><a href="#cb58-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.85</span>)<span class="sc">+</span></span>
<span id="cb58-33"><a href="#cb58-33" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb58-34"><a href="#cb58-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># facet_wrap(genotype~name, scales = "free") +</span></span>
<span id="cb58-35"><a href="#cb58-35" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ggh4x::facet_grid2(genotype~name, scales = "free_y", independent = "y") +</span></span>
<span id="cb58-36"><a href="#cb58-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb58-37"><a href="#cb58-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Time after transcription arrest"</span>) <span class="sc">+</span></span>
<span id="cb58-38"><a href="#cb58-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of PRM sites"</span>) <span class="sc">+</span></span>
<span id="cb58-39"><a href="#cb58-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="figure-4a-response-to-reviewer" class="level1">
<h1>Figure 4A response to reviewer:</h1>
<p>One of the reviewers correctly pointed out that figure 4A should show the proportion of each gene type modified relative to the number of expressed genes in that category. Do that here now that I’ve loaded in expression data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a> n_expressed <span class="ot">&lt;-</span> gene_types <span class="sc">%&gt;%</span>  </span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type2 =</span> <span class="fu">case_when</span>(type <span class="sc">==</span> <span class="st">"small_nuclear_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"small_nucleolar_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"antisense_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"antisense_long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"other_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"ribosomal_rna"</span> <span class="sc">~</span> <span class="st">"rRNA"</span>,</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"protein_coding"</span> <span class="sc">~</span> <span class="st">"mRNA"</span>,</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"pre_trna"</span> <span class="sc">~</span> <span class="st">"tRNA"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type2 <span class="sc">!=</span> <span class="st">"miRNA_primary_transcript"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>   <span class="fu">left_join</span>(., tpm, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>   <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>   dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>type) <span class="sc">%&gt;%</span></span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(gene, type2)) <span class="sc">%&gt;%</span></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., samples, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"name"</span> <span class="ot">=</span> <span class="st">"Run"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Experiment, gene, type2) <span class="sc">%&gt;%</span></span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_tpm =</span> <span class="fu">mean</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb59-21"><a href="#cb59-21" aria-hidden="true" tabindex="-1"></a>   <span class="fu">group_by</span>(Experiment, type2) <span class="sc">%&gt;%</span></span>
<span id="cb59-22"><a href="#cb59-22" aria-hidden="true" tabindex="-1"></a>   <span class="fu">summarise</span>(<span class="at">n_expressed =</span> <span class="fu">sum</span>(mean_tpm <span class="sc">&gt;=</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb59-23"><a href="#cb59-23" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb59-24"><a href="#cb59-24" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(Experiment <span class="sc">==</span> <span class="st">"WT-0"</span> <span class="sc">|</span> Experiment <span class="sc">==</span> <span class="st">"WT-480"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-25"><a href="#cb59-25" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">time2 =</span> <span class="fu">case_when</span>(Experiment <span class="sc">==</span> <span class="st">"WT-0"</span> <span class="sc">~</span> <span class="st">"Time point: 0"</span>,</span>
<span id="cb59-26"><a href="#cb59-26" aria-hidden="true" tabindex="-1"></a>                            Experiment <span class="sc">==</span> <span class="st">"WT-480"</span> <span class="sc">~</span> <span class="st">"Time point: 8hr"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Experiment', 'gene'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'Experiment'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a> deg <span class="sc">%&gt;%</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(full_position, gene, Experiment) <span class="sc">%&gt;%</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., gene_types, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Experiment, type) <span class="sc">%&gt;%</span></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_mod_genes =</span> <span class="fu">n_distinct</span>(gene)) <span class="sc">%&gt;%</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Experiment,</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"time"</span>),</span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> time <span class="sc">==</span> <span class="dv">480</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type2 =</span> <span class="fu">case_when</span>(type <span class="sc">==</span> <span class="st">"small_nuclear_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb61-15"><a href="#cb61-15" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"small_nucleolar_rna"</span> <span class="sc">~</span> <span class="st">"sn/snoRNA"</span>,</span>
<span id="cb61-16"><a href="#cb61-16" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"antisense_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb61-17"><a href="#cb61-17" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"antisense_long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb61-18"><a href="#cb61-18" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"long_noncoding_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb61-19"><a href="#cb61-19" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"other_rna"</span> <span class="sc">~</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb61-20"><a href="#cb61-20" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"ribosomal_rna"</span> <span class="sc">~</span> <span class="st">"rRNA"</span>,</span>
<span id="cb61-21"><a href="#cb61-21" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"protein_coding"</span> <span class="sc">~</span> <span class="st">"mRNA"</span>,</span>
<span id="cb61-22"><a href="#cb61-22" aria-hidden="true" tabindex="-1"></a>                           type <span class="sc">==</span> <span class="st">"pre_trna"</span> <span class="sc">~</span> <span class="st">"tRNA"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb61-23"><a href="#cb61-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type2 <span class="sc">!=</span> <span class="st">"miRNA_primary_transcript"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-24"><a href="#cb61-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">time2 =</span> <span class="fu">case_when</span>(time <span class="sc">==</span> <span class="st">"0"</span> <span class="sc">~</span> <span class="st">"Time point: 0"</span>,</span>
<span id="cb61-25"><a href="#cb61-25" aria-hidden="true" tabindex="-1"></a>                           time <span class="sc">==</span> <span class="st">"480"</span> <span class="sc">~</span> <span class="st">"Time point: 8hr"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb61-26"><a href="#cb61-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(type2, time2) <span class="sc">%&gt;%</span></span>
<span id="cb61-27"><a href="#cb61-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_genes =</span> <span class="fu">sum</span>(n_mod_genes)) <span class="sc">%&gt;%</span></span>
<span id="cb61-28"><a href="#cb61-28" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-29"><a href="#cb61-29" aria-hidden="true" tabindex="-1"></a>   <span class="fu">left_join</span>(., n_expressed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"type2"</span>, <span class="st">"time2"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb61-30"><a href="#cb61-30" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">prop =</span> <span class="dv">100</span><span class="sc">*</span>(n_genes <span class="sc">/</span> n_expressed)) <span class="sc">%&gt;%</span></span>
<span id="cb61-31"><a href="#cb61-31" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">prop2 =</span> <span class="fu">case_when</span>(prop <span class="sc">&gt;</span> <span class="dv">100</span> <span class="sc">~</span> <span class="dv">100</span>,</span>
<span id="cb61-32"><a href="#cb61-32" aria-hidden="true" tabindex="-1"></a>                            prop <span class="sc">&lt;=</span> <span class="dv">100</span> <span class="sc">~</span> prop)) <span class="sc">%&gt;%</span></span>
<span id="cb61-33"><a href="#cb61-33" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">prop3 =</span> <span class="fu">round</span>(prop2, <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb61-34"><a href="#cb61-34" aria-hidden="true" tabindex="-1"></a>   <span class="fu">filter</span>(type2 <span class="sc">!=</span> <span class="st">"rRNA"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-35"><a href="#cb61-35" aria-hidden="true" tabindex="-1"></a>     <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> type2, <span class="at">x =</span> prop2, <span class="at">fill =</span> type2)) <span class="sc">+</span></span>
<span id="cb61-36"><a href="#cb61-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb61-37"><a href="#cb61-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> prop3),<span class="at">size =</span> <span class="dv">12</span>, <span class="at">hjust =</span> <span class="sc">-</span>.<span class="dv">05</span>) <span class="sc">+</span></span>
<span id="cb61-38"><a href="#cb61-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_y_log10() +</span></span>
<span id="cb61-39"><a href="#cb61-39" aria-hidden="true" tabindex="-1"></a> <span class="co"># coord_flip() +</span></span>
<span id="cb61-40"><a href="#cb61-40" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb61-41"><a href="#cb61-41" aria-hidden="true" tabindex="-1"></a>   <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb61-42"><a href="#cb61-42" aria-hidden="true" tabindex="-1"></a>   <span class="co">#scale_y_discrete(breaks = c(0,1,2,3,4,5)) +</span></span>
<span id="cb61-43"><a href="#cb61-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>time2) <span class="sc">+</span></span>
<span id="cb61-44"><a href="#cb61-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb61-45"><a href="#cb61-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Percent of group modified"</span>) <span class="sc">+</span></span>
<span id="cb61-46"><a href="#cb61-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb61-47"><a href="#cb61-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb61-48"><a href="#cb61-48" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Experiment'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'type2'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a> <span class="fu">ggsave</span>(<span class="at">plot =</span> <span class="fu">last_plot</span>(),</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>        <span class="st">"/Users/kylepalos/Desktop/Nelson_lab/epitranscriptomics/mini_comparative_project/dec_2023_figures/figure4/percent_gene_type_modified.pdf"</span>,</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">9</span>, <span class="at">units =</span> <span class="st">"in"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Analyze authors’ RPKM decreases in PRM marked mRNAs vs non-marked mRNAs</p>
<p>This code generates figures:</p>
</section>
<section id="figure-4c" class="level1">
<h1>Figure 4C</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pivot RPKM longer</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>rpkm <span class="ot">&lt;-</span> rpkm <span class="sc">%&gt;%</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">!</span>gene, </span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>,<span class="st">"time"</span>), </span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_pattern =</span> <span class="st">"(WT|sov|varicose|double)_(.*)"</span>,</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"value"</span>) </span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>rpkm<span class="sc">$</span>time <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_mean"</span>, <span class="st">""</span>, rpkm<span class="sc">$</span>time)</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>rpkm<span class="sc">$</span>time <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(rpkm<span class="sc">$</span>time)</span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-12"><a href="#cb64-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-13"><a href="#cb64-13" aria-hidden="true" tabindex="-1"></a><span class="co"># establish which mRNAs/genes are modified at TP0 </span></span>
<span id="cb64-14"><a href="#cb64-14" aria-hidden="true" tabindex="-1"></a><span class="co"># or after TP0</span></span>
<span id="cb64-15"><a href="#cb64-15" aria-hidden="true" tabindex="-1"></a><span class="co"># or modified in any experiemnt</span></span>
<span id="cb64-16"><a href="#cb64-16" aria-hidden="true" tabindex="-1"></a><span class="co"># or never observed to be modified</span></span>
<span id="cb64-17"><a href="#cb64-17" aria-hidden="true" tabindex="-1"></a>mod_0_genes <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb64-18"><a href="#cb64-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-19"><a href="#cb64-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb64-20"><a href="#cb64-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class =</span> <span class="st">"tp_0_mod"</span>)</span>
<span id="cb64-21"><a href="#cb64-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-22"><a href="#cb64-22" aria-hidden="true" tabindex="-1"></a>mod_after_0 <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb64-23"><a href="#cb64-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(., mod_0_genes, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-24"><a href="#cb64-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb64-25"><a href="#cb64-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class =</span> <span class="st">"after_0_mod"</span>)</span>
<span id="cb64-26"><a href="#cb64-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-27"><a href="#cb64-27" aria-hidden="true" tabindex="-1"></a>mod_genes <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb64-28"><a href="#cb64-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene)</span>
<span id="cb64-29"><a href="#cb64-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-30"><a href="#cb64-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-31"><a href="#cb64-31" aria-hidden="true" tabindex="-1"></a>never_mod <span class="ot">&lt;-</span> rpkm <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb64-32"><a href="#cb64-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(., mod_genes, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-33"><a href="#cb64-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class =</span> <span class="st">"never_mod"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-34"><a href="#cb64-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., rpkm, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb64-35"><a href="#cb64-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-36"><a href="#cb64-36" aria-hidden="true" tabindex="-1"></a>mod_0_genes <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb64-37"><a href="#cb64-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-38"><a href="#cb64-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb64-39"><a href="#cb64-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class =</span> <span class="st">"tp_0_mod"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-40"><a href="#cb64-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., rpkm, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb64-41"><a href="#cb64-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-42"><a href="#cb64-42" aria-hidden="true" tabindex="-1"></a>mod_after_0 <span class="ot">&lt;-</span> deg <span class="sc">%&gt;%</span></span>
<span id="cb64-43"><a href="#cb64-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(., mod_0_genes, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-44"><a href="#cb64-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb64-45"><a href="#cb64-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class =</span> <span class="st">"after_0_mod"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-46"><a href="#cb64-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., rpkm, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb64-47"><a href="#cb64-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-48"><a href="#cb64-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-49"><a href="#cb64-49" aria-hidden="true" tabindex="-1"></a><span class="co"># let's just compare modified at TP0 vs never mod</span></span>
<span id="cb64-50"><a href="#cb64-50" aria-hidden="true" tabindex="-1"></a>tp0_vs_never_mod <span class="ot">&lt;-</span> <span class="fu">rbind</span>(mod_0_genes, never_mod)</span>
<span id="cb64-51"><a href="#cb64-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-52"><a href="#cb64-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-53"><a href="#cb64-53" aria-hidden="true" tabindex="-1"></a><span class="co"># summarize the dataframe for number of expressed genes remaining at each time point relative to TP0</span></span>
<span id="cb64-54"><a href="#cb64-54" aria-hidden="true" tabindex="-1"></a>tp0_vs_never_mod_decay <span class="ot">&lt;-</span> tp0_vs_never_mod <span class="sc">%&gt;%</span></span>
<span id="cb64-55"><a href="#cb64-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(class, genotype, time) <span class="sc">%&gt;%</span></span>
<span id="cb64-56"><a href="#cb64-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_expressed =</span> <span class="fu">sum</span>(value <span class="sc">&gt;=</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-57"><a href="#cb64-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-58"><a href="#cb64-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-59"><a href="#cb64-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(class, genotype) <span class="sc">%&gt;%</span></span>
<span id="cb64-60"><a href="#cb64-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percentage_remaining =</span> n_expressed <span class="sc">/</span> n_expressed[time <span class="sc">==</span> <span class="dv">0</span>] <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-61"><a href="#cb64-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'class', 'genotype'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># factor and re-level</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>tp0_vs_never_mod_decay<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">factor</span>(tp0_vs_never_mod_decay<span class="sc">$</span>genotype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"WT"</span>, <span class="st">"sov"</span>, <span class="st">"varicose"</span>, <span class="st">"double"</span>))</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>tp0_vs_never_mod_decay <span class="sc">%&gt;%</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class2 =</span> <span class="fu">case_when</span>(class <span class="sc">==</span> <span class="st">"never_mod"</span> <span class="sc">~</span> <span class="st">"Never modified"</span>,</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>                            class <span class="sc">==</span> <span class="st">"tp_0_mod"</span> <span class="sc">~</span> <span class="st">"Mod at 0TP"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(time), <span class="at">y =</span> percentage_remaining, <span class="at">color =</span> genotype,</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> class2)) <span class="sc">+</span></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">2.75</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  facet_wrap(~ class2) +</span></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Percent genes expressed"</span>) <span class="sc">+</span></span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Time after transcription arrest"</span>) <span class="sc">+</span></span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size =</span>  <span class="fu">unit</span>(<span class="fl">0.75</span>, <span class="st">"in"</span>)) <span class="sc">+</span></span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">linetype=</span><span class="st">"class"</span>, <span class="at">color =</span> <span class="st">"genotype"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Next make the figure for how RPKM decreases over time</p>
</section>
<section id="figure-4d" class="level1">
<h1>Figure 4D</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> tp0_vs_never_mod <span class="sc">%&gt;%</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class2 =</span> <span class="fu">case_when</span>(class <span class="sc">==</span> <span class="st">"never_mod"</span> <span class="sc">~</span> <span class="st">"Never modified"</span>,</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>                            class <span class="sc">==</span> <span class="st">"tp_0_mod"</span> <span class="sc">~</span> <span class="st">"Mod at 0TP"</span>,</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>                            class <span class="sc">==</span> <span class="st">"after_0_mod"</span> <span class="sc">~</span> <span class="st">"Mod after 0TP"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene, class2, genotype) <span class="sc">%&gt;%</span></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">decrease =</span> ((value <span class="sc">-</span> value[time<span class="sc">==</span><span class="dv">0</span>])<span class="sc">/</span> value[time <span class="sc">==</span> <span class="dv">0</span>])<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">!=</span> <span class="st">"0"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(decrease <span class="sc">&lt;=</span> <span class="dv">100</span> <span class="sc">&amp;</span> decrease <span class="sc">&gt;=</span> <span class="sc">-</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(class2, time) <span class="sc">%&gt;%</span></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">median</span>(decrease, <span class="at">na.rm =</span>T))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'class2'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>tp0_vs_never_mod <span class="sc">%&gt;%</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">class2 =</span> <span class="fu">case_when</span>(class <span class="sc">==</span> <span class="st">"never_mod"</span> <span class="sc">~</span> <span class="st">"Never modified"</span>,</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>                            class <span class="sc">==</span> <span class="st">"tp_0_mod"</span> <span class="sc">~</span> <span class="st">"Mod at 0TP"</span>,</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>                            class <span class="sc">==</span> <span class="st">"after_0_mod"</span> <span class="sc">~</span> <span class="st">"Mod after 0TP"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene, class2, genotype) <span class="sc">%&gt;%</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">decrease =</span> ((value <span class="sc">-</span> value[time<span class="sc">==</span><span class="dv">0</span>])<span class="sc">/</span> value[time <span class="sc">==</span> <span class="dv">0</span>])<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"WT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(time <span class="sc">!=</span> <span class="st">"0"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a> <span class="co"># filter(decrease &lt;= 100 &amp; decrease &gt;= -100) %&gt;%</span></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(time), <span class="at">y =</span> decrease, <span class="at">fill =</span> class2)) <span class="sc">+</span></span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">name =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">100</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ggh4x::facet_grid2(time ~ class2, scales = "free_y", independent = "y") + </span></span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>)<span class="sc">+</span></span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_y_log10() +</span></span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>)) <span class="sc">+</span></span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Time after transcription arrest"</span>) <span class="sc">+</span></span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"RNA abundance % change after TP0"</span>) <span class="sc">+</span></span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a>             <span class="at">linewidth =</span> <span class="fl">1.45</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1918 rows containing non-finite outside the scale range
(`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Finally, generate</p>
</section>
<section id="supplemental-figure-9c" class="level1">
<h1>Supplemental Figure 9C</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="ot">&lt;-</span> tpm_sum2 <span class="sc">%&gt;%</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., mod_summary, <span class="at">by =</span> <span class="st">"Experiment"</span>)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="ot">&lt;-</span> summary_combined <span class="sc">%&gt;%</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Experiment,</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"time"</span>),</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"n expr genes"</span> <span class="ot">=</span> <span class="dv">4</span>,</span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>                <span class="st">"n modified RNAs"</span> <span class="ot">=</span> <span class="dv">5</span>,</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>                <span class="st">"n modified sites"</span> <span class="ot">=</span> <span class="dv">6</span>)</span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>summary_combined<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">factor</span>(summary_combined<span class="sc">$</span>genotype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"WT"</span>, <span class="st">"sov"</span>, <span class="st">"varicose"</span>, <span class="st">"double"</span>))</span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a>summary_combined <span class="sc">%&gt;%</span></span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">"protein_coding"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_norm =</span> <span class="st">`</span><span class="at">n modified RNAs</span><span class="st">`</span><span class="sc">/</span><span class="st">`</span><span class="at">n expr genes</span><span class="st">`</span>,</span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">density =</span> <span class="st">`</span><span class="at">n modified sites</span><span class="st">`</span><span class="sc">/</span><span class="st">`</span><span class="at">n modified RNAs</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"n modified normalized"</span> <span class="ot">=</span> gene_norm,</span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">mod_density =</span> density) <span class="sc">%&gt;%</span></span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>type) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb71-23"><a href="#cb71-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(genotype,time)) <span class="sc">%&gt;%</span></span>
<span id="cb71-24"><a href="#cb71-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(name <span class="sc">==</span> <span class="st">"n modified normalized"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-25"><a href="#cb71-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(time), <span class="at">y =</span> value, <span class="at">color =</span> genotype)) <span class="sc">+</span></span>
<span id="cb71-26"><a href="#cb71-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">3.85</span>)<span class="sc">+</span></span>
<span id="cb71-27"><a href="#cb71-27" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb71-28"><a href="#cb71-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># facet_wrap(genotype~name, scales = "free") +</span></span>
<span id="cb71-29"><a href="#cb71-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ggh4x::facet_grid2(genotype~name, scales = "free_y", independent = "y") +</span></span>
<span id="cb71-30"><a href="#cb71-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb71-31"><a href="#cb71-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Time after transcription arrest"</span>) <span class="sc">+</span></span>
<span id="cb71-32"><a href="#cb71-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#facet_wrap(~genotype, ncol = 2) +</span></span>
<span id="cb71-33"><a href="#cb71-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Proportion of modified RNAs"</span>) <span class="sc">+</span></span>
<span id="cb71-34"><a href="#cb71-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="section" class="level1">
<h1></h1>
<section id="now-start-working-on-supplemental-figure-10-nonsense-mediated-decay" class="level2">
<h2 class="anchored" data-anchor-id="now-start-working-on-supplemental-figure-10-nonsense-mediated-decay">Now start working on supplemental figure 10 (nonsense mediated decay)</h2>
<p>Start by loading in the counts of the SMG7/PAD4 RNA-seq dataset</p>
<p>Keep in mind, these counts were taken from salmon and rounded to integers - for the paper I just used tximport to load in “true raw” values into DESeq2</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load in counts</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/smg7_raw_salmon_counts.txt?raw=TRUE"</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/smg7_counts.txt"</span>)</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/smg7_counts.txt"</span>,</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> counts <span class="sc">%&gt;%</span></span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"gene"</span>)</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a><span class="co"># make metadata table for DESeq2</span></span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">colnames</span>(counts) <span class="sc">%&gt;%</span></span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Run =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"pad4"</span>, <span class="st">"smg7_pad4"</span>), <span class="at">each =</span> <span class="dv">3</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>perform differential expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,
    tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    second, second&lt;-</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:lubridate':

    %within%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biobase'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:MatrixGenerics':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(counts, metadata, <span class="sc">~</span> genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"genotype"</span>, <span class="st">"smg7_pad4"</span>, <span class="st">"pad4"</span>),</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">tidy =</span> T)</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>up <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> .<span class="dv">05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(log2FoldChange <span class="sc">&gt;=</span> <span class="dv">1</span>)</span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>down <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> .<span class="dv">05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(log2FoldChange <span class="sc">&lt;=</span> <span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load in ModTect for this dataset</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/45581586"</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/smg7_modtect.txt.gz"</span>)</span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/smg7_modtect.txt.gz"</span>, </span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>                            <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">trim_ws =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(depth <span class="sc">&gt;=</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 441315 Columns: 45
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (11): chrom, reference_nt, gene_chr, gene, code, strand, source, type, o...
dbl (34): position, position2, mono-alleleLogP, bi-alleleLogP, tri-alleleLog...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>Do data cleaning and determine if there’s a significant overlap between mRNAs up-regulated in smg7_pad4/pad4 and mRNAs marked by PRMs</p>
</section>
</section>
<section id="supplemental-figure-10a" class="level1">
<h1>Supplemental Figure 10A</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>mods<span class="sc">$</span>file <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_intersected.txt"</span>, <span class="st">""</span>, mods<span class="sc">$</span>file)</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"file"</span> <span class="ot">=</span> <span class="st">"Run"</span>))</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>mods<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gene:"</span>, <span class="st">""</span>, mods<span class="sc">$</span>gene)</span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"full_position"</span>, chrom<span class="sc">:</span>position2, <span class="at">sep =</span> <span class="st">"-"</span>) </span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(full_position) <span class="sc">%&gt;%</span></span>
<span id="cb119-13"><a href="#cb119-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_exps =</span> <span class="fu">n_distinct</span>(file)) <span class="sc">%&gt;%</span></span>
<span id="cb119-14"><a href="#cb119-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb119-15"><a href="#cb119-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_exps <span class="sc">&gt;=</span> <span class="dv">3</span>)</span>
<span id="cb119-16"><a href="#cb119-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-17"><a href="#cb119-17" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb119-18"><a href="#cb119-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., keep, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb119-19"><a href="#cb119-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-20"><a href="#cb119-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-21"><a href="#cb119-21" aria-hidden="true" tabindex="-1"></a>wt_mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span> </span>
<span id="cb119-22"><a href="#cb119-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"pad4"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb119-23"><a href="#cb119-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene)</span>
<span id="cb119-24"><a href="#cb119-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-25"><a href="#cb119-25" aria-hidden="true" tabindex="-1"></a>smg_mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span> </span>
<span id="cb119-26"><a href="#cb119-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"smg7_pad4"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb119-27"><a href="#cb119-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene)</span>
<span id="cb119-28"><a href="#cb119-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-29"><a href="#cb119-29" aria-hidden="true" tabindex="-1"></a>all_mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb119-30"><a href="#cb119-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene)</span>
<span id="cb119-31"><a href="#cb119-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-32"><a href="#cb119-32" aria-hidden="true" tabindex="-1"></a>mod_overlap <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">`</span><span class="at">PRMs in smg7</span><span class="st">`</span> <span class="ot">=</span> smg_mods<span class="sc">$</span>gene,</span>
<span id="cb119-33"><a href="#cb119-33" aria-hidden="true" tabindex="-1"></a>                    <span class="st">`</span><span class="at">smg7 up-regulated</span><span class="st">`</span> <span class="ot">=</span> up<span class="sc">$</span>row)</span>
<span id="cb119-34"><a href="#cb119-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-35"><a href="#cb119-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-36"><a href="#cb119-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplotify)</span>
<span id="cb119-37"><a href="#cb119-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggvenn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>venn <span class="ot">&lt;-</span> <span class="fu">as.ggplot</span>(<span class="fu">ggvenn</span>(mod_overlap, <span class="at">text_size =</span> <span class="dv">12</span>,,</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">show_percentage =</span> F,</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">fill_color =</span> <span class="fu">c</span>(<span class="st">"blue"</span>,<span class="st">"red2"</span>),</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">set_name_size =</span> <span class="dv">8</span>),</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>venn</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The numbers a slightly off because, as I mentioned above, I provided the rounded salmon raw counts instead of the raw salmon files which would have been more unruly to host and download.</p>
<p>Next, download the PRMs closest to start/stop codons and exon-exon junctions for determining whether PRMs are enriched at specific features in NMD mutants:</p>
<p>These “closest” files were generated by using bedtools intersect on a file of start codons, stop codons, and exon-exon junction (splice sites)</p>
<p>Additionally, download an isoform level TPM file so we can remove non-expressed isoforms.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># first load in isoform level expression</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/smg7_transcript_tpm.txt?raw=TRUE"</span></span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/smg7_pad4_tx_tpm.txt"</span>)</span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>tx_tpm <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/smg7_pad4_tx_tpm.txt"</span>, </span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 53467 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (1): tx
dbl (6): SRR1192255, SRR1192256, SRR1192257, SRR1192258, SRR1192259, SRR1192260

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate a list of isoforms that are expressed</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>expressed_txs <span class="ot">&lt;-</span> tx_tpm <span class="sc">%&gt;%</span></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>tx) <span class="sc">%&gt;%</span></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(tx) <span class="sc">%&gt;%</span></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_exp_expr =</span> <span class="fu">sum</span>(value <span class="sc">&gt;</span> .<span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_exp_expr <span class="sc">&gt;=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(tx)</span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-13"><a href="#cb124-13" aria-hidden="true" tabindex="-1"></a><span class="co"># download Araport11 GTF to establish a transcript to gene mapping file and the strand of genes for orienting PRMs to exon-exon junctions:</span></span>
<span id="cb124-14"><a href="#cb124-14" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/48355528"</span></span>
<span id="cb124-15"><a href="#cb124-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-16"><a href="#cb124-16" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/araport.gtf.gz"</span>)</span>
<span id="cb124-17"><a href="#cb124-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-18"><a href="#cb124-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicFeatures)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi

Attaching package: 'AnnotationDbi'

The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> <span class="fu">makeTxDbFromGFF</span>(<span class="st">"~/Downloads/araport.gtf.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in call_fun_in_txdbmaker("makeTxDbFromGFF", ...): makeTxDbFromGFF() has moved to the txdbmaker package. Please call
  txdbmaker::makeTxDbFromGFF() to get rid of this warning.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Import genomic features from the file as a GRanges object ... OK
Prepare the 'metadata' data frame ... OK
Make the TxDb object ... </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type
  exon. This information was ignored.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>OK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="fu">keys</span>(txdb, <span class="at">keytype =</span> <span class="st">"TXNAME"</span>)</span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>tx2gene <span class="ot">&lt;-</span> <span class="fu">select</span>(txdb, k, <span class="st">"GENEID"</span>, <span class="st">"TXNAME"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tx2gene is a transcript to gene mapping file</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="co"># get the strand of all genes</span></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">genes</span>(txdb)</span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>gene_strand <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(genes) <span class="sc">%&gt;%</span></span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strand, gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> gene_id)</span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-10"><a href="#cb133-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-11"><a href="#cb133-11" aria-hidden="true" tabindex="-1"></a><span class="co"># now load in the bedtools outputs of PRMs in the 2 different genotypes</span></span>
<span id="cb133-12"><a href="#cb133-12" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/smg_pad4_closest_feature.txt?raw=TRUE"</span></span>
<span id="cb133-13"><a href="#cb133-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-14"><a href="#cb133-14" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/smg7_pad4_close.txt"</span>)</span>
<span id="cb133-15"><a href="#cb133-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-16"><a href="#cb133-16" aria-hidden="true" tabindex="-1"></a>smg_close <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/smg7_pad4_close.txt"</span>, </span>
<span id="cb133-17"><a href="#cb133-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb133-18"><a href="#cb133-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb133-19"><a href="#cb133-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., expressed_txs, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X9"</span> <span class="ot">=</span> <span class="st">"tx"</span>))  <span class="sc">%&gt;%</span> <span class="co"># only retain expressed transcripts</span></span>
<span id="cb133-20"><a href="#cb133-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"[.]1"</span>, X9)) <span class="co"># only retain primary isoform</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 115623 Columns: 11
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (5): X1, X4, X6, X9, X10
dbl (6): X2, X3, X5, X7, X8, X11

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure4_files/pad4_closest_feature.txt?raw=TRUE"</span></span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/pad4_close.txt"</span>)</span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a>pad_close <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/pad4_close.txt"</span>, </span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col_names =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., expressed_txs, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X9"</span> <span class="ot">=</span> <span class="st">"tx"</span>))  <span class="sc">%&gt;%</span> <span class="co"># only retain expressed transcripts</span></span>
<span id="cb135-9"><a href="#cb135-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"[.]1"</span>, X9)) <span class="co"># only retain primary isoform</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 109983 Columns: 11
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (5): X1, X4, X6, X9, X10
dbl (6): X2, X3, X5, X7, X8, X11

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>next block of code will wrangle the “close” data frames to generate</p>
</section>
<section id="supplemental-figure-10b" class="level1">
<h1>Supplemental Figure 10B:</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># join with tx2gene, and rename some labels</span></span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>pad_close <span class="ot">&lt;-</span> pad_close <span class="sc">%&gt;%</span></span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X9"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(X10 <span class="sc">!=</span> <span class="st">"."</span>) <span class="sc">%&gt;%</span></span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(X4 <span class="sc">==</span> GENEID) <span class="sc">%&gt;%</span></span>
<span id="cb137-6"><a href="#cb137-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#left_join(., strand, by = "gene") %&gt;%</span></span>
<span id="cb137-7"><a href="#cb137-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb137-8"><a href="#cb137-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">new_label =</span> <span class="fu">case_when</span>(X10 <span class="sc">==</span> <span class="st">"stop_codon"</span> <span class="sc">~</span> <span class="st">"Stop codon"</span>,</span>
<span id="cb137-9"><a href="#cb137-9" aria-hidden="true" tabindex="-1"></a>                               X10 <span class="sc">==</span> <span class="st">"5ej"</span> <span class="sc">~</span> <span class="st">"5' exon junction"</span>,</span>
<span id="cb137-10"><a href="#cb137-10" aria-hidden="true" tabindex="-1"></a>                               X10 <span class="sc">==</span> <span class="st">"3ej"</span> <span class="sc">~</span> <span class="st">"3' exon junction"</span>,</span>
<span id="cb137-11"><a href="#cb137-11" aria-hidden="true" tabindex="-1"></a>                               X10 <span class="sc">==</span> <span class="st">"start_codon"</span> <span class="sc">~</span> <span class="st">"Start codon"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb137-12"><a href="#cb137-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., gene_strand, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X4"</span> <span class="ot">=</span> <span class="st">"gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb137-13"><a href="#cb137-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">new_distance =</span> <span class="fu">ifelse</span>(strand <span class="sc">==</span> <span class="st">'+'</span>, <span class="sc">-</span>X11, X11))</span>
<span id="cb137-14"><a href="#cb137-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-15"><a href="#cb137-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-16"><a href="#cb137-16" aria-hidden="true" tabindex="-1"></a>smg_close <span class="ot">&lt;-</span> smg_close <span class="sc">%&gt;%</span></span>
<span id="cb137-17"><a href="#cb137-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X9"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb137-18"><a href="#cb137-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(X10 <span class="sc">!=</span> <span class="st">"."</span>) <span class="sc">%&gt;%</span></span>
<span id="cb137-19"><a href="#cb137-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(X4 <span class="sc">==</span> GENEID) <span class="sc">%&gt;%</span></span>
<span id="cb137-20"><a href="#cb137-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#left_join(., strand, by = "gene") %&gt;%</span></span>
<span id="cb137-21"><a href="#cb137-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb137-22"><a href="#cb137-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">new_label =</span> <span class="fu">case_when</span>(X10 <span class="sc">==</span> <span class="st">"stop_codon"</span> <span class="sc">~</span> <span class="st">"Stop codon"</span>,</span>
<span id="cb137-23"><a href="#cb137-23" aria-hidden="true" tabindex="-1"></a>                               X10 <span class="sc">==</span> <span class="st">"5ej"</span> <span class="sc">~</span> <span class="st">"5' exon junction"</span>,</span>
<span id="cb137-24"><a href="#cb137-24" aria-hidden="true" tabindex="-1"></a>                               X10 <span class="sc">==</span> <span class="st">"3ej"</span> <span class="sc">~</span> <span class="st">"3' exon junction"</span>,</span>
<span id="cb137-25"><a href="#cb137-25" aria-hidden="true" tabindex="-1"></a>                               X10 <span class="sc">==</span> <span class="st">"start_codon"</span> <span class="sc">~</span> <span class="st">"Start codon"</span>))  <span class="sc">%&gt;%</span></span>
<span id="cb137-26"><a href="#cb137-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., gene_strand, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X4"</span> <span class="ot">=</span> <span class="st">"gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb137-27"><a href="#cb137-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">new_distance =</span> <span class="fu">ifelse</span>(strand <span class="sc">==</span> <span class="st">'+'</span>, <span class="sc">-</span>X11, X11))</span>
<span id="cb137-28"><a href="#cb137-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-29"><a href="#cb137-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-30"><a href="#cb137-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-31"><a href="#cb137-31" aria-hidden="true" tabindex="-1"></a>facet_label <span class="ot">&lt;-</span> smg_close <span class="sc">%&gt;%</span></span>
<span id="cb137-32"><a href="#cb137-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(new_label) <span class="sc">%&gt;%</span></span>
<span id="cb137-33"><a href="#cb137-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">x =</span> <span class="sc">-</span><span class="dv">160</span>,</span>
<span id="cb137-34"><a href="#cb137-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> .<span class="dv">02</span>)</span>
<span id="cb137-35"><a href="#cb137-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-36"><a href="#cb137-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-37"><a href="#cb137-37" aria-hidden="true" tabindex="-1"></a>smg_close<span class="sc">$</span>new_label <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(smg_close<span class="sc">$</span>new_label)</span>
<span id="cb137-38"><a href="#cb137-38" aria-hidden="true" tabindex="-1"></a>pad_close<span class="sc">$</span>new_label <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pad_close<span class="sc">$</span>new_label)</span>
<span id="cb137-39"><a href="#cb137-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-40"><a href="#cb137-40" aria-hidden="true" tabindex="-1"></a>smg_close<span class="sc">$</span>new_label <span class="ot">&lt;-</span> <span class="fu">fct_relevel</span>(smg_close<span class="sc">$</span>new_label, <span class="fu">c</span>(</span>
<span id="cb137-41"><a href="#cb137-41" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Start codon"</span>, <span class="st">"5' exon junction"</span>, <span class="st">"3' exon junction"</span>, <span class="st">"Stop codon"</span></span>
<span id="cb137-42"><a href="#cb137-42" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb137-43"><a href="#cb137-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-44"><a href="#cb137-44" aria-hidden="true" tabindex="-1"></a>pad_close<span class="sc">$</span>new_label <span class="ot">&lt;-</span> <span class="fu">fct_relevel</span>(pad_close<span class="sc">$</span>new_label, <span class="fu">c</span>(</span>
<span id="cb137-45"><a href="#cb137-45" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Start codon"</span>, <span class="st">"5' exon junction"</span>, <span class="st">"3' exon junction"</span>, <span class="st">"Stop codon"</span></span>
<span id="cb137-46"><a href="#cb137-46" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb137-47"><a href="#cb137-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-48"><a href="#cb137-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-49"><a href="#cb137-49" aria-hidden="true" tabindex="-1"></a>smg_close <span class="ot">&lt;-</span> smg_close <span class="sc">%&gt;%</span></span>
<span id="cb137-50"><a href="#cb137-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">"smg7"</span>)</span>
<span id="cb137-51"><a href="#cb137-51" aria-hidden="true" tabindex="-1"></a>pad_close <span class="ot">&lt;-</span> pad_close <span class="sc">%&gt;%</span></span>
<span id="cb137-52"><a href="#cb137-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="st">"pad4"</span>)</span>
<span id="cb137-53"><a href="#cb137-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-54"><a href="#cb137-54" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> <span class="fu">rbind</span>(smg_close, pad_close) </span>
<span id="cb137-55"><a href="#cb137-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-56"><a href="#cb137-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-57"><a href="#cb137-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-58"><a href="#cb137-58" aria-hidden="true" tabindex="-1"></a>combined <span class="sc">%&gt;%</span></span>
<span id="cb137-59"><a href="#cb137-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype2 =</span> <span class="fu">case_when</span>(genotype <span class="sc">==</span> <span class="st">"smg7"</span> <span class="sc">~</span> <span class="st">"smg7-pad4"</span>,</span>
<span id="cb137-60"><a href="#cb137-60" aria-hidden="true" tabindex="-1"></a>                              genotype <span class="sc">==</span> <span class="st">"pad4"</span> <span class="sc">~</span> <span class="st">"pad4"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb137-61"><a href="#cb137-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> new_distance)) <span class="sc">+</span></span>
<span id="cb137-62"><a href="#cb137-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">n =</span> <span class="dv">65536</span>, <span class="at">bw =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb137-63"><a href="#cb137-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">150</span>,<span class="dv">150</span>))<span class="sc">+</span></span>
<span id="cb137-64"><a href="#cb137-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"reverse"</span>) <span class="sc">+</span></span>
<span id="cb137-65"><a href="#cb137-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(genotype2 <span class="sc">~</span> new_label) <span class="sc">+</span></span>
<span id="cb137-66"><a href="#cb137-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb137-67"><a href="#cb137-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb137-68"><a href="#cb137-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Modification density'</span>) <span class="sc">+</span></span>
<span id="cb137-69"><a href="#cb137-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Nucleotides surrounding feature"</span>) <span class="sc">+</span></span>
<span id="cb137-70"><a href="#cb137-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb137-71"><a href="#cb137-71" aria-hidden="true" tabindex="-1"></a>        <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Going to do a similar workflow here to plot the PRM build up these transcript features with mRNAs up-regulated in smg7 (putative targets of NMD) vs mRNAs not significantly affected in the smg7 genotype</p>
<p>generates</p>
</section>
<section id="supplemental-figure-10c" class="level1">
<h1>Supplemental Figure 10C</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>smg7_combined_not_de <span class="ot">&lt;-</span> combined <span class="sc">%&gt;%</span></span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype2 =</span> <span class="fu">case_when</span>(genotype <span class="sc">==</span> <span class="st">"smg7"</span> <span class="sc">~</span> <span class="st">"smg7-pad4"</span>,</span>
<span id="cb139-3"><a href="#cb139-3" aria-hidden="true" tabindex="-1"></a>                               genotype <span class="sc">==</span> <span class="st">"pad4"</span> <span class="sc">~</span> <span class="st">"pad4"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb139-4"><a href="#cb139-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"smg7"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb139-5"><a href="#cb139-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(., up, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X4"</span> <span class="ot">=</span> <span class="st">"row"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb139-6"><a href="#cb139-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DE =</span> <span class="st">"no"</span>)</span>
<span id="cb139-7"><a href="#cb139-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-8"><a href="#cb139-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-9"><a href="#cb139-9" aria-hidden="true" tabindex="-1"></a>smg7_combined_de <span class="ot">&lt;-</span> combined <span class="sc">%&gt;%</span></span>
<span id="cb139-10"><a href="#cb139-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype2 =</span> <span class="fu">case_when</span>(genotype <span class="sc">==</span> <span class="st">"smg7"</span> <span class="sc">~</span> <span class="st">"smg7-pad4"</span>,</span>
<span id="cb139-11"><a href="#cb139-11" aria-hidden="true" tabindex="-1"></a>                               genotype <span class="sc">==</span> <span class="st">"pad4"</span> <span class="sc">~</span> <span class="st">"pad4"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb139-12"><a href="#cb139-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genotype <span class="sc">==</span> <span class="st">"smg7"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb139-13"><a href="#cb139-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., up, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"X4"</span> <span class="ot">=</span> <span class="st">"row"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb139-14"><a href="#cb139-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">18</span><span class="sc">:</span><span class="dv">23</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb139-15"><a href="#cb139-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DE =</span> <span class="st">"yes"</span>)</span>
<span id="cb139-16"><a href="#cb139-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-17"><a href="#cb139-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-18"><a href="#cb139-18" aria-hidden="true" tabindex="-1"></a>smg7_combined <span class="ot">&lt;-</span> <span class="fu">rbind</span>(smg7_combined_de, smg7_combined_not_de)</span>
<span id="cb139-19"><a href="#cb139-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-20"><a href="#cb139-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-21"><a href="#cb139-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-22"><a href="#cb139-22" aria-hidden="true" tabindex="-1"></a>smg7_combined <span class="sc">%&gt;%</span></span>
<span id="cb139-23"><a href="#cb139-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DE2 =</span> <span class="fu">case_when</span>(DE <span class="sc">==</span> <span class="st">"yes"</span> <span class="sc">~</span> <span class="st">"up in smg7"</span>,</span>
<span id="cb139-24"><a href="#cb139-24" aria-hidden="true" tabindex="-1"></a>                          DE <span class="sc">==</span> <span class="st">"no"</span> <span class="sc">~</span> <span class="st">"no difference"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb139-25"><a href="#cb139-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> new_distance)) <span class="sc">+</span></span>
<span id="cb139-26"><a href="#cb139-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_histogram(aes(y=..density..), alpha=0.75, </span></span>
<span id="cb139-27"><a href="#cb139-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                 position="identity", binwidth = 2,</span></span>
<span id="cb139-28"><a href="#cb139-28" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                 color = "black", fill = "white")+</span></span>
<span id="cb139-29"><a href="#cb139-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">n =</span> <span class="dv">65536</span>, <span class="at">bw =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb139-30"><a href="#cb139-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_density(size = 1.5,bw = 1, aes(color = genotype)) +</span></span>
<span id="cb139-31"><a href="#cb139-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">150</span>,<span class="dv">150</span>))<span class="sc">+</span></span>
<span id="cb139-32"><a href="#cb139-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"reverse"</span>) <span class="sc">+</span></span>
<span id="cb139-33"><a href="#cb139-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_text(data = facet_label, aes(label = new_label, x = x, y = y),</span></span>
<span id="cb139-34"><a href="#cb139-34" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           hjust = 0, fontface=2, size = 4.5) +</span></span>
<span id="cb139-35"><a href="#cb139-35" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_color_npg</span>(<span class="at">name =</span> <span class="st">"smg7 DE?"</span>) <span class="sc">+</span></span>
<span id="cb139-36"><a href="#cb139-36" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_y_log10()+</span></span>
<span id="cb139-37"><a href="#cb139-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(DE2<span class="sc">~</span>new_label) <span class="sc">+</span></span>
<span id="cb139-38"><a href="#cb139-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb139-39"><a href="#cb139-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb139-40"><a href="#cb139-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Modification density'</span>) <span class="sc">+</span></span>
<span id="cb139-41"><a href="#cb139-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Nucleotides surrounding feature"</span>) <span class="sc">+</span></span>
<span id="cb139-42"><a href="#cb139-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb139-43"><a href="#cb139-43" aria-hidden="true" tabindex="-1"></a>        <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Finally, we need to determine the overlap of mRNAs predicted to be NMD sensitive vs PRM marked mRNAs in the Arabidopsis tissue expression atlas:</p>
<p>start by loading in tissue expression atlas PRMs and associated TPM values</p>
<p>This will generate</p>
</section>
<section id="supplemental-figure-10d" class="level1">
<h1>Supplemental Figure 10D</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/45632223"</span></span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/at_expression_atlas_prms.txt.gz"</span>)</span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-5"><a href="#cb140-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:S4Vectors':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    between, first, last</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a>modtect <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/at_expression_atlas_prms.txt.gz"</span>,</span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col_names =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb149-4"><a href="#cb149-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(file,</span>
<span id="cb149-5"><a href="#cb149-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"SRA"</span>, <span class="st">"other"</span>),</span>
<span id="cb149-6"><a href="#cb149-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb149-7"><a href="#cb149-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>other) <span class="sc">%&gt;%</span></span>
<span id="cb149-8"><a href="#cb149-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(depth <span class="sc">&gt;=</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 394777 Columns: 45</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (11): chrom, reference_nt, gene_chr, gene, code, strand, source, type, o...
dbl (34): position, position2, mono-alleleLogP, bi-alleleLogP, tri-alleleLog...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># download and load in sample metadata</span></span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure2_files/at_expression_atlas_metadata.xlsx?raw=TRUE"</span></span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/at_expression_atlas_metadata.xlsx"</span>)</span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb152-7"><a href="#cb152-7" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"~/Downloads/at_expression_atlas_metadata.xlsx"</span>)</span>
<span id="cb152-8"><a href="#cb152-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-9"><a href="#cb152-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-10"><a href="#cb152-10" aria-hidden="true" tabindex="-1"></a>modtect <span class="ot">&lt;-</span> modtect <span class="sc">%&gt;%</span></span>
<span id="cb152-11"><a href="#cb152-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"SRA"</span> <span class="ot">=</span> <span class="st">"Run"</span>))</span>
<span id="cb152-12"><a href="#cb152-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-13"><a href="#cb152-13" aria-hidden="true" tabindex="-1"></a><span class="co"># for sorghum, i kept mods that were present in at least 3 replicates across the whole dataset</span></span>
<span id="cb152-14"><a href="#cb152-14" aria-hidden="true" tabindex="-1"></a>sites_to_keep <span class="ot">&lt;-</span> modtect <span class="sc">%&gt;%</span></span>
<span id="cb152-15"><a href="#cb152-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> <span class="st">"full_position"</span>, <span class="fu">c</span>(chrom<span class="sc">:</span>position2), <span class="at">remove =</span> F, <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-16"><a href="#cb152-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position, SRA) <span class="sc">%&gt;%</span></span>
<span id="cb152-17"><a href="#cb152-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(full_position) <span class="sc">%&gt;%</span></span>
<span id="cb152-18"><a href="#cb152-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">occurences_per_samples =</span> <span class="fu">n_distinct</span>(SRA)) <span class="sc">%&gt;%</span> </span>
<span id="cb152-19"><a href="#cb152-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(occurences_per_samples <span class="sc">&gt;=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-20"><a href="#cb152-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position)</span>
<span id="cb152-21"><a href="#cb152-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-22"><a href="#cb152-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-23"><a href="#cb152-23" aria-hidden="true" tabindex="-1"></a>modtect <span class="ot">&lt;-</span> modtect <span class="sc">%&gt;%</span></span>
<span id="cb152-24"><a href="#cb152-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F, <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-25"><a href="#cb152-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., sites_to_keep, <span class="at">by =</span> <span class="st">"full_position"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-26"><a href="#cb152-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() </span>
<span id="cb152-27"><a href="#cb152-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-28"><a href="#cb152-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-29"><a href="#cb152-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 394519 rows to 53,939 rows</span></span>
<span id="cb152-30"><a href="#cb152-30" aria-hidden="true" tabindex="-1"></a>modtect<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"gene:"</span>, <span class="st">""</span>, modtect<span class="sc">$</span>gene)</span>
<span id="cb152-31"><a href="#cb152-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-32"><a href="#cb152-32" aria-hidden="true" tabindex="-1"></a><span class="co"># download and read in TPM</span></span>
<span id="cb152-33"><a href="#cb152-33" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure2_files/at_tissue_expression_atlas_tpm.txt?raw=TRUE"</span></span>
<span id="cb152-34"><a href="#cb152-34" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/at_expression_atlas_tpm.txt"</span>)</span>
<span id="cb152-35"><a href="#cb152-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-36"><a href="#cb152-36" aria-hidden="true" tabindex="-1"></a>tpm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/at_expression_atlas_tpm.txt"</span>,</span>
<span id="cb152-37"><a href="#cb152-37" aria-hidden="true" tabindex="-1"></a>                  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> T)</span>
<span id="cb152-38"><a href="#cb152-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-39"><a href="#cb152-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-40"><a href="#cb152-40" aria-hidden="true" tabindex="-1"></a><span class="co"># get expressed genes</span></span>
<span id="cb152-41"><a href="#cb152-41" aria-hidden="true" tabindex="-1"></a>expressed <span class="ot">&lt;-</span> tpm <span class="sc">%&gt;%</span></span>
<span id="cb152-42"><a href="#cb152-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb152-43"><a href="#cb152-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb152-44"><a href="#cb152-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">max =</span> <span class="fu">max</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb152-45"><a href="#cb152-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb152-46"><a href="#cb152-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(max <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-47"><a href="#cb152-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(gene)</span>
<span id="cb152-48"><a href="#cb152-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-49"><a href="#cb152-49" aria-hidden="true" tabindex="-1"></a><span class="co"># 27401  expressed genes</span></span>
<span id="cb152-50"><a href="#cb152-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-51"><a href="#cb152-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-52"><a href="#cb152-52" aria-hidden="true" tabindex="-1"></a><span class="co"># to do the overlap, predict NMD mRNA targets using the factR package</span></span>
<span id="cb152-53"><a href="#cb152-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factR)</span>
<span id="cb152-54"><a href="#cb152-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-55"><a href="#cb152-55" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span> <span class="fu">importGTF</span>(<span class="st">"~/Downloads/araport.gtf.gz"</span>)</span>
<span id="cb152-56"><a href="#cb152-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-57"><a href="#cb152-57" aria-hidden="true" tabindex="-1"></a>nmd_predict <span class="ot">&lt;-</span> <span class="fu">predictNMD</span>(gtf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Predicting NMD sensitivities for 48238 mRNAs</code></pre>
</div>
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a>nmd_genes <span class="ot">&lt;-</span> nmd_predict <span class="sc">%&gt;%</span></span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a> dplyr<span class="sc">::</span><span class="fu">filter</span>(is_NMD <span class="sc">==</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"transcript"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GENEID)</span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a>prm_genes <span class="ot">&lt;-</span> modtect <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(gene)</span>
<span id="cb154-8"><a href="#cb154-8" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">`</span><span class="at">Mergner et al PRMs</span><span class="st">`</span> <span class="ot">=</span> prm_genes<span class="sc">$</span>gene,</span>
<span id="cb154-9"><a href="#cb154-9" aria-hidden="true" tabindex="-1"></a>                <span class="st">`</span><span class="at">Predicted NMD</span><span class="st">`</span> <span class="ot">=</span> nmd_genes<span class="sc">$</span>GENEID)</span>
<span id="cb154-10"><a href="#cb154-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggvenn</span>(overlap)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure4_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>