<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.313">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kyle Palos">

<title>Figure 3</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="figure3_files/libs/clipboard/clipboard.min.js"></script>
<script src="figure3_files/libs/quarto-html/quarto.js"></script>
<script src="figure3_files/libs/quarto-html/popper.min.js"></script>
<script src="figure3_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="figure3_files/libs/quarto-html/anchor.min.js"></script>
<link href="figure3_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="figure3_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="figure3_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="figure3_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="figure3_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Figure 3</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Kyle Palos </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<p>This script performs the analyses going into Figure 3 and associated supplement of the comparative PRM manuscript</p>
<p>An important input for these analyses is the output of Bedtools closest using filtered PRM sites and a BED file of start codons, stop codons, and splice junctions (with both the 5’ and 3’ edge of the junction annotated)</p>
<p>Specifically, these analyses focus on Sorghum seedling PRMs and their distance to relevant transcript features (i.e., those mentioned above).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'

The following objects are masked from 'package:lubridate':

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from 'package:dplyr':

    between, first, last

The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'

The following object is masked from 'package:purrr':

    discard

The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
</div>
<p>Download the BEDTools closest output:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure3_files/sorghum_mods_closest_feature.txt?raw=TRUE"</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/sorghum_closest.txt"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>closest <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/sorghum_closest.txt"</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">header =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get a transcript to gene mapping file</p>
<p>Download GFF3 annotation for Sorghum from Ensembl plants and use GenomicFeatures to get the mapping file we need:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-58/gff3/sorghum_bicolor/Sorghum_bicolor.Sorghum_bicolor_NCBIv3.58.gff3.gz"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/sorghum_annotation.gff3.gz"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicFeatures)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,
    tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:data.table':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:lubridate':

    second, second&lt;-</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:data.table':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:lubridate':

    %within%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> <span class="fu">makeTxDbFromGFF</span>(<span class="st">"~/Downloads/sorghum_annotation.gff3.gz"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in call_fun_in_txdbmaker("makeTxDbFromGFF", ...): makeTxDbFromGFF() has moved to the txdbmaker package. Please call
  txdbmaker::makeTxDbFromGFF() to get rid of this warning.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Import genomic features from the file as a GRanges object ...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>OK</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Prepare the 'metadata' data frame ... OK
Make the TxDb object ... </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .extract_exons_from_GRanges(exon_IDX, gr, mcols0, tx_IDX, feature = "exon", : 262 exons couldn't be linked to a transcript so were dropped (showing
  only the first 6):
  seqid   start     end strand   ID               Name
1     1  562993  563102      + &lt;NA&gt; ENSRNA049997130-E1
2     1 1019419 1019521      - &lt;NA&gt; ENSRNA049997137-E1
3     1 1026702 1026839      - &lt;NA&gt; ENSRNA049469069-E1
4     1 2876710 2876888      + &lt;NA&gt; ENSRNA049997145-E1
5     1 4354010 4354105      + &lt;NA&gt; ENSRNA049997152-E1
6     1 5218433 5218532      - &lt;NA&gt; ENSRNA049997158-E1
                         Parent Parent_type
1 transcript:ENSRNA049997130-T1        &lt;NA&gt;
2 transcript:ENSRNA049997137-T1        &lt;NA&gt;
3 transcript:ENSRNA049469069-T1        &lt;NA&gt;
4 transcript:ENSRNA049997145-T1        &lt;NA&gt;
5 transcript:ENSRNA049997152-T1        &lt;NA&gt;
6 transcript:ENSRNA049997158-T1        &lt;NA&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>OK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="fu">keys</span>(txdb, <span class="at">keytype =</span> <span class="st">"TXNAME"</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>tx2gene <span class="ot">&lt;-</span> <span class="fu">select</span>(txdb, k, <span class="st">"GENEID"</span>, <span class="st">"TXNAME"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
</div>
<p>Perform some data wrangling to clean up the “closest” data frame for plotting and analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>closest <span class="ot">&lt;-</span> closest <span class="sc">%&gt;%</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(V4, <span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"pred_mod"</span>), <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"V9"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(GENEID, <span class="at">.before =</span> V9)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>closest<span class="sc">$</span>V10 <span class="ot">&lt;-</span> <span class="fu">fct_relevel</span>(closest<span class="sc">$</span>V10, <span class="st">"start_codon"</span>, <span class="st">"5ej"</span>, <span class="st">"3ej"</span>, <span class="st">"stop_codon"</span>)</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>closest2 <span class="ot">&lt;-</span> closest <span class="sc">%&gt;%</span></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(V10 <span class="sc">!=</span> <span class="st">"."</span>) <span class="sc">%&gt;%</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">==</span> GENEID) <span class="sc">%&gt;%</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">new_label =</span> <span class="fu">case_when</span>(V10 <span class="sc">==</span> <span class="st">"stop_codon"</span> <span class="sc">~</span> <span class="st">"Stop codon"</span>,</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>                               V10 <span class="sc">==</span> <span class="st">"5ej"</span> <span class="sc">~</span> <span class="st">"5' exon junction"</span>,</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>                               V10 <span class="sc">==</span> <span class="st">"3ej"</span> <span class="sc">~</span> <span class="st">"3' exon junction"</span>,</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>                               V10 <span class="sc">==</span> <span class="st">"start_codon"</span> <span class="sc">~</span> <span class="st">"Start codon"</span>)) </span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>closest2<span class="sc">$</span>new_label <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(closest2<span class="sc">$</span>new_label)</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>closest2<span class="sc">$</span>new_label <span class="ot">&lt;-</span> <span class="fu">fct_relevel</span>(closest2<span class="sc">$</span>new_label, <span class="fu">c</span>(</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Start codon"</span>, <span class="st">"5' exon junction"</span>, <span class="st">"3' exon junction"</span>, <span class="st">"Stop codon"</span></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load in isoform level TPM to remove isoforms not expressed</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure3_files/sorghum_transcript_tpm.txt?raw=TRUE"</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/sorghum_transcript_tpm.txt"</span>)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>tx_tpm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/sorghum_transcript_tpm.txt"</span>,</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>tx_expressed <span class="ot">&lt;-</span> tx_tpm <span class="sc">%&gt;%</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tpm_avg =</span> <span class="fu">mean</span>(<span class="fu">c</span>(sorghum1, sorghum2))) <span class="sc">%&gt;%</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(tpm_avg <span class="sc">&gt;=</span> <span class="dv">1</span>)</span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="co"># get other transcript information from the TXDB file:</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>tx_info <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">transcripts</span>(txdb, <span class="at">columns =</span> <span class="st">"tx_name"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Transcript strand has to be taken into account in the context of bedtools closest. Ideally, look in a genome browser to ensure that the genomic coordinate will be the correct positive or “negative” distance from the feature of interest</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># establish the strand of all transcripts and join it with the corresponding gene</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>sb_strand <span class="ot">&lt;-</span> tx_info <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">5</span>,<span class="dv">6</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">strand =</span> <span class="dv">1</span>, <span class="at">tx =</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"tx"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>))</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="co"># remove isoforms not expressed and add on the strand</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>closest2 <span class="ot">&lt;-</span> closest2 <span class="sc">%&gt;%</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., tx_expressed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"V9"</span> <span class="ot">=</span> <span class="st">"tx"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., sb_strand, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"V9"</span> <span class="ot">=</span> <span class="st">"tx"</span>,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"GENEID"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="co"># manually specifying where some text is going to go in the figure</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>facet_label <span class="ot">&lt;-</span> closest2 <span class="sc">%&gt;%</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(new_label) <span class="sc">%&gt;%</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">x =</span> <span class="sc">-</span><span class="dv">160</span>,</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> .<span class="dv">028</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot and make</p>
<section id="figure-3d" class="level1">
<h1>Figure 3D:</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># write the mods that are close to junctions</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>closest2 <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">new_distance =</span> <span class="fu">case_when</span>(strand <span class="sc">==</span> <span class="st">"-"</span> <span class="sc">~</span> V11,</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>                                  strand <span class="sc">==</span> <span class="st">"+"</span> <span class="sc">~</span> <span class="sc">-</span>V11)) <span class="sc">%&gt;%</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(V10 <span class="sc">==</span> <span class="st">"5ej"</span> <span class="sc">|</span> V10 <span class="sc">==</span> <span class="st">"3ej"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(new_distance) <span class="sc">&lt;=</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(.,</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>              <span class="st">"~/Downloads/ej_mods.bed"</span>,</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">row.names =</span> F, <span class="at">col.names =</span> F, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> F)</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a><span class="co"># For positive strand genes, flip the sign of the distance value to make it consistent</span></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>closest2 <span class="sc">%&gt;%</span></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">new_distance =</span> <span class="fu">case_when</span>(strand <span class="sc">==</span> <span class="st">"-"</span> <span class="sc">~</span> V11,</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>                                  strand <span class="sc">==</span> <span class="st">"+"</span> <span class="sc">~</span> <span class="sc">-</span>V11)) <span class="sc">%&gt;%</span></span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> new_distance)) <span class="sc">+</span></span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y=</span>..density..), <span class="at">alpha=</span><span class="fl">0.75</span>, </span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">binwidth =</span> <span class="dv">2</span>,</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a> <span class="fu">geom_density</span>(<span class="at">size =</span> .<span class="dv">75</span>, <span class="at">n =</span> <span class="dv">65536</span>, <span class="at">bw =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">150</span>,<span class="dv">150</span>))<span class="sc">+</span></span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"reverse"</span>,</span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">100</span>, <span class="dv">0</span>, <span class="dv">100</span>),</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"-100 (5')"</span>,</span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"0"</span>,</span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"100 (3')"</span>)) <span class="sc">+</span></span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> facet_label, <span class="fu">aes</span>(<span class="at">label =</span> new_label, <span class="at">x =</span> x, <span class="at">y =</span> y),</span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">fontface=</span><span class="dv">2</span>, <span class="at">size =</span> <span class="dv">7</span>) <span class="sc">+</span></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a> <span class="co"># scale_y_log10()+</span></span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a> <span class="fu">facet_wrap</span>(<span class="sc">~</span> new_label, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Modification density'</span>) <span class="sc">+</span></span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Nucleotides surrounding feature"</span>) <span class="sc">+</span></span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="fl">0.00</span>, <span class="fl">0.025</span>),</span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"0"</span>,<span class="st">".025"</span>)) <span class="sc">+</span></span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The dot-dot notation (`..density..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(density)` instead.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure3_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Next, to generate Figure 3E, we need to annotate which exons of a transcript are the first, last, internal, etc.</p>
<p>Use the exons by function from GenomicFeatures to accomplish this</p>
<p>This is actually partially encoded by the exon rank column but I did not utilize it previously.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>exons <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">exonsBy</span>(txdb, <span class="at">by =</span> <span class="st">"tx"</span>, <span class="at">use.names =</span> T))</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> exons <span class="sc">%&gt;%</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(strand <span class="sc">==</span> <span class="st">"+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(group_name) <span class="sc">%&gt;%</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon_n =</span> <span class="fu">row_number</span>(),</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_exons =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(total_exons <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon_position =</span> <span class="fu">case_when</span>(exon_n <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"first"</span>,</span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>                                   exon_n <span class="sc">==</span> total_exons <span class="sc">~</span> <span class="st">"last"</span>,</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>                                   <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"internal"</span>))</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="co"># for negative strand transcripts, it's a little different</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>neg <span class="ot">&lt;-</span> exons <span class="sc">%&gt;%</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand <span class="sc">==</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(group_name) <span class="sc">%&gt;%</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(start), <span class="at">.by_group =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(group_name) <span class="sc">%&gt;%</span></span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon_n =</span> <span class="fu">row_number</span>(),</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">total_exons =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(total_exons <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon_position =</span> <span class="fu">case_when</span>(exon_n <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"first"</span>,</span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>                                   exon_n <span class="sc">==</span> total_exons <span class="sc">~</span> <span class="st">"last"</span>,</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a>                                   <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"internal"</span>)) </span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a>exons2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(pos,neg)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Additionally, you will need a bedtools intersection of RNA modifications against exons of all mRNAs</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure3_files/sorghum_mod_exon_intersect.txt?raw=TRUE"</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/sorghum_mod_exon_intersection.txt"</span>)</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>int <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/sorghum_mod_exon_intersection.txt"</span>, </span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">escape_double =</span> <span class="cn">FALSE</span>, </span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">col_names =</span> <span class="cn">FALSE</span>, <span class="at">trim_ws =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 10059 Columns: 12
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (5): X1, X4, X6, X9, X11
dbl (7): X2, X3, X5, X7, X8, X10, X12

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>clean up intersection file and add some additional labels to the “first, internal, and last” exon file, then join them together</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># establish a more specific "first internal" and "last internal" exon (i.e., the second exon and penultimate exon, respectively)</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>exons3 <span class="ot">&lt;-</span> exons2 <span class="sc">%&gt;%</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"group_name"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> <span class="dv">3</span>, <span class="at">start =</span> <span class="dv">4</span>, <span class="at">end =</span> <span class="dv">5</span>, <span class="at">exon_length =</span> <span class="dv">6</span>, <span class="at">strand =</span> <span class="dv">7</span>,</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">exon_number =</span> <span class="dv">11</span>, <span class="at">n_exons =</span> <span class="dv">12</span>, <span class="at">position =</span> <span class="dv">13</span>, <span class="at">gene =</span> <span class="dv">14</span>, <span class="at">tx =</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_exons <span class="sc">&gt;=</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">position2 =</span> <span class="fu">case_when</span>(exon_number <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"first_internal"</span>,</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>                               n_exons <span class="sc">-</span> exon_number <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"last_internal"</span>,</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a>                               <span class="cn">TRUE</span> <span class="sc">~</span> position)) <span class="sc">%&gt;%</span></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a><span class="co"># load in TPM from figure 1</span></span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>sb_tpm <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/sorghum_tpm.txt?raw=TRUE"</span></span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Download</span></span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(sb_tpm, <span class="at">destfile =</span> <span class="st">"~/Downloads/sb_tpm.txt"</span>)</span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-23"><a href="#cb55-23" aria-hidden="true" tabindex="-1"></a>sb_tpm <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/sb_tpm.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col_names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 34779 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (1): Gene
dbl (1): tpm_avg

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># retain expressed genes</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>sb_expressed <span class="ot">&lt;-</span> sb_tpm <span class="sc">%&gt;%</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(tpm_avg <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Gene)</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a><span class="co"># only keep exons of expressed genes and compute the total amount of exonic space taken up by each exon type and proportionally how much space that is across all exons</span></span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>exon_summary <span class="ot">&lt;-</span> exons3 <span class="sc">%&gt;%</span></span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., sb_expressed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span> <span class="co"># only assess expressed genes</span></span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(position2) <span class="sc">%&gt;%</span></span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">total_space =</span> <span class="fu">sum</span>(exon_length)) <span class="sc">%&gt;%</span></span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fractional_space =</span> (total_space<span class="sc">/</span><span class="fu">sum</span>(total_space)))</span>
<span id="cb57-14"><a href="#cb57-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-15"><a href="#cb57-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-16"><a href="#cb57-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-17"><a href="#cb57-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Now we clean up the BedTools intersect file</span></span>
<span id="cb57-18"><a href="#cb57-18" aria-hidden="true" tabindex="-1"></a>int_clean <span class="ot">&lt;-</span> int  <span class="sc">%&gt;%</span></span>
<span id="cb57-19"><a href="#cb57-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">chr =</span> <span class="dv">1</span>, <span class="at">start =</span> <span class="dv">2</span>, <span class="at">end =</span> <span class="dv">3</span>, <span class="at">gene_mod =</span> <span class="dv">4</span>, <span class="at">var_prop =</span> <span class="dv">5</span>, <span class="at">gene_chr =</span> <span class="dv">6</span>, <span class="at">gene_start =</span> <span class="dv">7</span>, <span class="at">gene_end =</span> <span class="dv">8</span>, <span class="at">gene =</span> <span class="dv">9</span>, <span class="at">score =</span> <span class="dv">10</span>, <span class="at">strand =</span> <span class="dv">11</span>, <span class="at">overlap =</span> <span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-20"><a href="#cb57-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(gene_mod, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"mod_gene"</span>, <span class="st">"mod"</span>), <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-21"><a href="#cb57-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb57-22"><a href="#cb57-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">tx =</span> gene,</span>
<span id="cb57-23"><a href="#cb57-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">gene =</span> GENEID) <span class="sc">%&gt;%</span></span>
<span id="cb57-24"><a href="#cb57-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(gene, <span class="at">.before =</span> tx) <span class="sc">%&gt;%</span></span>
<span id="cb57-25"><a href="#cb57-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(mod_gene <span class="sc">==</span> gene) <span class="sc">%&gt;%</span></span>
<span id="cb57-26"><a href="#cb57-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>score, <span class="sc">-</span>overlap) <span class="sc">%&gt;%</span></span>
<span id="cb57-27"><a href="#cb57-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">exon_chr =</span> gene_chr,</span>
<span id="cb57-28"><a href="#cb57-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">exon_start =</span> gene_start,</span>
<span id="cb57-29"><a href="#cb57-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">exon_end =</span> gene_end,</span>
<span id="cb57-30"><a href="#cb57-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">exon_gene =</span> gene) <span class="sc">%&gt;%</span></span>
<span id="cb57-31"><a href="#cb57-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"exon_tx"</span>, exon_gene<span class="sc">:</span>tx, <span class="at">sep =</span> <span class="st">"-"</span>, <span class="at">remove =</span> F)</span>
<span id="cb57-32"><a href="#cb57-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-33"><a href="#cb57-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-34"><a href="#cb57-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-35"><a href="#cb57-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Now, join together RNA modifications and only retain expressed isoforms</span></span>
<span id="cb57-36"><a href="#cb57-36" aria-hidden="true" tabindex="-1"></a>join <span class="ot">&lt;-</span> int_clean <span class="sc">%&gt;%</span></span>
<span id="cb57-37"><a href="#cb57-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon_start =</span> exon_start <span class="sc">+</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="co"># account for off by one BED 0-based errors</span></span>
<span id="cb57-38"><a href="#cb57-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., exons3, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"exon_chr"</span> <span class="ot">=</span> <span class="st">"chr"</span>,</span>
<span id="cb57-39"><a href="#cb57-39" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"exon_start"</span> <span class="ot">=</span> <span class="st">"start"</span>,</span>
<span id="cb57-40"><a href="#cb57-40" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"exon_end"</span> <span class="ot">=</span> <span class="st">"end"</span>,</span>
<span id="cb57-41"><a href="#cb57-41" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"exon_gene"</span> <span class="ot">=</span> <span class="st">"gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb57-42"><a href="#cb57-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, mod, mod_gene, position2, exon_length, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb57-43"><a href="#cb57-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-44"><a href="#cb57-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., tx_expressed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"tx.x"</span> <span class="ot">=</span> <span class="st">"tx"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., exons3, by = c(exon_chr = "chr", exon_start = "start", : Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 19 of `x` matches multiple rows in `y`.
ℹ Row 101805 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
</div>
<p>Ok, now we can finally make</p>
</section>
<section id="figure-3e" class="level1">
<h1>Figure 3E</h1>
<p>For the first exon junction edge</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get mods for first exon 3' edge</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="co"># positive strand</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>first_3edge_pos <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"first"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_3ej =</span> exon_end <span class="sc">-</span> start) </span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a><span class="co"># negative strand</span></span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>first_3edge_neg <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"first"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_3ej =</span> start <span class="sc">-</span> exon_start) </span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>first_3edge <span class="ot">&lt;-</span> <span class="fu">rbind</span>(first_3edge_pos,</span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>                     first_3edge_neg)</span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-21"><a href="#cb59-21" aria-hidden="true" tabindex="-1"></a><span class="co"># now get mods for second exon 5' edge (first internal exon)</span></span>
<span id="cb59-22"><a href="#cb59-22" aria-hidden="true" tabindex="-1"></a><span class="co"># positive strand</span></span>
<span id="cb59-23"><a href="#cb59-23" aria-hidden="true" tabindex="-1"></a>second_5edge_pos <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb59-24"><a href="#cb59-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"first_internal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-25"><a href="#cb59-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-26"><a href="#cb59-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb59-27"><a href="#cb59-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_5ej =</span> start <span class="sc">-</span> exon_start) </span>
<span id="cb59-28"><a href="#cb59-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-29"><a href="#cb59-29" aria-hidden="true" tabindex="-1"></a><span class="co"># negative strand</span></span>
<span id="cb59-30"><a href="#cb59-30" aria-hidden="true" tabindex="-1"></a>second_5edge_neg <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb59-31"><a href="#cb59-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"first_internal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-32"><a href="#cb59-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-33"><a href="#cb59-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb59-34"><a href="#cb59-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_5ej =</span> exon_end <span class="sc">-</span> start) </span>
<span id="cb59-35"><a href="#cb59-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-36"><a href="#cb59-36" aria-hidden="true" tabindex="-1"></a>second_5edge <span class="ot">&lt;-</span> <span class="fu">rbind</span>(second_5edge_pos,</span>
<span id="cb59-37"><a href="#cb59-37" aria-hidden="true" tabindex="-1"></a>                      second_5edge_neg)</span>
<span id="cb59-38"><a href="#cb59-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-39"><a href="#cb59-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-40"><a href="#cb59-40" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> first_3edge <span class="sc">%&gt;%</span></span>
<span id="cb59-41"><a href="#cb59-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span>distance_to_3ej)) <span class="sc">+</span></span>
<span id="cb59-42"><a href="#cb59-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"black"</span>) <span class="sc">+</span> <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb59-43"><a href="#cb59-43" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb59-44"><a href="#cb59-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb59-45"><a href="#cb59-45" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ggtitle("Mod distribution in the FIRST exon approaching 3' edge",</span></span>
<span id="cb59-46"><a href="#cb59-46" aria-hidden="true" tabindex="-1"></a>     <span class="co">#     subtitle = "Approaching first exon junction") +</span></span>
<span id="cb59-47"><a href="#cb59-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb59-48"><a href="#cb59-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb59-49"><a href="#cb59-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">15</span>,<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb59-50"><a href="#cb59-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span>
<span id="cb59-51"><a href="#cb59-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-52"><a href="#cb59-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-53"><a href="#cb59-53" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> second_5edge <span class="sc">%&gt;%</span></span>
<span id="cb59-54"><a href="#cb59-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> distance_to_5ej)) <span class="sc">+</span></span>
<span id="cb59-55"><a href="#cb59-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"black"</span>) <span class="sc">+</span> <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>),</span>
<span id="cb59-56"><a href="#cb59-56" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb59-57"><a href="#cb59-57" aria-hidden="true" tabindex="-1"></a> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb59-58"><a href="#cb59-58" aria-hidden="true" tabindex="-1"></a> <span class="co"># ggtitle("Mod distribution in the Second exon approaching 5' edge",</span></span>
<span id="cb59-59"><a href="#cb59-59" aria-hidden="true" tabindex="-1"></a>     <span class="co">#     subtitle = "Approaching first exon junction") +</span></span>
<span id="cb59-60"><a href="#cb59-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb59-61"><a href="#cb59-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">position =</span> <span class="st">"right"</span>,</span>
<span id="cb59-62"><a href="#cb59-62" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">15</span>,<span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb59-63"><a href="#cb59-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb59-64"><a href="#cb59-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>and for the last exon junction edge:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get mods for last internal exon 3' edge</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="co"># positive strand</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>last_internal_3edge_pos <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"last_internal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_3ej =</span> exon_end <span class="sc">-</span> start) </span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a><span class="co"># negative strand</span></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>last_internal_3edge_neg <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"last_internal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_3ej =</span> start <span class="sc">-</span> exon_start) </span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a>last_internal_3edge <span class="ot">&lt;-</span> <span class="fu">rbind</span>(last_internal_3edge_pos,</span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a>                     last_internal_3edge_neg)</span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-20"><a href="#cb60-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-21"><a href="#cb60-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-22"><a href="#cb60-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-23"><a href="#cb60-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-24"><a href="#cb60-24" aria-hidden="true" tabindex="-1"></a><span class="co"># now get mods for last exon 5' edge</span></span>
<span id="cb60-25"><a href="#cb60-25" aria-hidden="true" tabindex="-1"></a><span class="co"># positive strand</span></span>
<span id="cb60-26"><a href="#cb60-26" aria-hidden="true" tabindex="-1"></a>last_5edge_pos <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb60-27"><a href="#cb60-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-28"><a href="#cb60-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-29"><a href="#cb60-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb60-30"><a href="#cb60-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_5ej =</span> start <span class="sc">-</span> exon_start) </span>
<span id="cb60-31"><a href="#cb60-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-32"><a href="#cb60-32" aria-hidden="true" tabindex="-1"></a><span class="co"># negative strand</span></span>
<span id="cb60-33"><a href="#cb60-33" aria-hidden="true" tabindex="-1"></a>last_5edge_neg <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb60-34"><a href="#cb60-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(position2 <span class="sc">==</span> <span class="st">"last"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-35"><a href="#cb60-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(strand.x <span class="sc">==</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-36"><a href="#cb60-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(chr, start, end, position2, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb60-37"><a href="#cb60-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">distance_to_5ej =</span> exon_end <span class="sc">-</span> start) </span>
<span id="cb60-38"><a href="#cb60-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-39"><a href="#cb60-39" aria-hidden="true" tabindex="-1"></a>last_5edge <span class="ot">&lt;-</span> <span class="fu">rbind</span>(last_5edge_pos,</span>
<span id="cb60-40"><a href="#cb60-40" aria-hidden="true" tabindex="-1"></a>                      last_5edge_neg)</span>
<span id="cb60-41"><a href="#cb60-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-42"><a href="#cb60-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-43"><a href="#cb60-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-44"><a href="#cb60-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-45"><a href="#cb60-45" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> last_internal_3edge <span class="sc">%&gt;%</span></span>
<span id="cb60-46"><a href="#cb60-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span>distance_to_3ej)) <span class="sc">+</span></span>
<span id="cb60-47"><a href="#cb60-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"black"</span>) <span class="sc">+</span> <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">100</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb60-48"><a href="#cb60-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb60-49"><a href="#cb60-49" aria-hidden="true" tabindex="-1"></a> <span class="co"># ggtitle("Mod distribution in the LAST INTERNAL exon approaching 3' edge",</span></span>
<span id="cb60-50"><a href="#cb60-50" aria-hidden="true" tabindex="-1"></a>   <span class="co">#       subtitle = "Approaching last exon junction") +</span></span>
<span id="cb60-51"><a href="#cb60-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb60-52"><a href="#cb60-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb60-53"><a href="#cb60-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span>
<span id="cb60-54"><a href="#cb60-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-55"><a href="#cb60-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-56"><a href="#cb60-56" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> last_5edge  <span class="sc">%&gt;%</span></span>
<span id="cb60-57"><a href="#cb60-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> distance_to_5ej)) <span class="sc">+</span></span>
<span id="cb60-58"><a href="#cb60-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"black"</span>) <span class="sc">+</span> <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb60-59"><a href="#cb60-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb60-60"><a href="#cb60-60" aria-hidden="true" tabindex="-1"></a> <span class="co"># ggtitle("Mod distribution in the LAST exon approaching 5' edge",</span></span>
<span id="cb60-61"><a href="#cb60-61" aria-hidden="true" tabindex="-1"></a>    <span class="co">#      subtitle = "Approaching last exon junction") +</span></span>
<span id="cb60-62"><a href="#cb60-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb60-63"><a href="#cb60-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb60-64"><a href="#cb60-64" aria-hidden="true" tabindex="-1"></a> <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb60-65"><a href="#cb60-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Patch plots:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>(p1<span class="sc">+</span>p2) <span class="sc">/</span> </span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  (p3<span class="sc">+</span>p4) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure3_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Now, pivot for Figure 3A</p>
<p>The input data for 3A is the output of DeepTools computeMatrix with a bed file of RNA modifications in each species, specifically the RNA modifications from Figure 1.</p>
<p>This computeMatrix code would look like:</p>
<p>The input bigWig file is generated from the bed file (bed -&gt; bedGraph -&gt; bigWig)</p>
<pre><code>computeMatrix scale-regions \
-S setaria_mods_density.bw \
-R ~/Desktop/Nelson_lab/epitranscriptomics/genome_files/setaria_gffread.gtf \
--metagene -p 8 --averageTypeBins sum --missingDataAsZero --exonID CDS \
-a 500 -b 500 -o setaria_metagene.tab.gz</code></pre>
<p>Then, to get the data for R, you would do:</p>
<pre><code>plotProfile --matrixFile sorghum_metagene.tab.gz \
--outFileName sorghum.svg \
--outFileNameData sorghum_plot.tab \
--startLabel start --endLabel end</code></pre>
<p>The function for reading in this tab file and tidying it up for plotting comes from Jeff Grover:</p>
<p><a href="https://groverj3.github.io/articles/2019-06-28_making-better-metaplots-with-ggplot-part-2.html" class="uri">https://groverj3.github.io/articles/2019-06-28_making-better-metaplots-with-ggplot-part-2.html</a></p>
<p>Read in the tab files for plotting</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure3_files/maize_plot.tab?raw=TRUE"</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/maize_meta.txt"</span>)</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure3_files/setaria_plot.tab?raw=TRUE"</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/setaria_meta.txt"</span>)</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure3_files/sorghum_plot.tab?raw=TRUE"</span></span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/sorghum_meta.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Define the function by Jeff to read in the table and tidy it up for plotting:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>read_deeptools_table <span class="ot">&lt;-</span> <span class="cf">function</span>(file) {</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">count.fields</span>(file, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">readLines</span>(file)</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>  .splitvar <span class="ot">&lt;-</span> <span class="cf">function</span>(x, sep, n) {</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>    var <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(x, <span class="at">split =</span> sep))</span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>(var) <span class="ot">&lt;-</span> n</span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(var)</span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, <span class="fu">lapply</span>(x, .splitvar, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">n =</span> n))</span>
<span id="cb65-13"><a href="#cb65-13" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">apply</span>(x, <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span>
<span id="cb65-14"><a href="#cb65-14" aria-hidden="true" tabindex="-1"></a>  plot_table <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(<span class="fu">read.csv</span>(<span class="at">text =</span> x, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)[<span class="sc">-</span><span class="dv">1</span>,])  <span class="co"># Remove first row with "gene" label</span></span>
<span id="cb65-15"><a href="#cb65-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-16"><a href="#cb65-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(plot_table)</span>
<span id="cb65-17"><a href="#cb65-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Read them in now:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>maize <span class="ot">&lt;-</span> <span class="fu">read_deeptools_table</span>(<span class="st">'~/Downloads/maize_meta.txt'</span>)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>maize_long <span class="ot">&lt;-</span> <span class="fu">gather</span>(maize, <span class="st">'sample'</span>, <span class="st">'score'</span>, <span class="sc">-</span>bin.labels, <span class="sc">-</span>bins)</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>maize_long<span class="sc">$</span>bin.labels <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"start"</span>, <span class="st">"start codon"</span>, maize_long<span class="sc">$</span>bin.labels)</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>maize_long<span class="sc">$</span>bin.labels <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"end"</span>, <span class="st">"stop codon"</span>, maize_long<span class="sc">$</span>bin.labels)</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>setaria <span class="ot">&lt;-</span> <span class="fu">read_deeptools_table</span>(<span class="st">'~/Downloads/setaria_meta.txt'</span>)</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>setaria_long <span class="ot">&lt;-</span> <span class="fu">gather</span>(setaria, <span class="st">'sample'</span>, <span class="st">'score'</span>, <span class="sc">-</span>bin.labels, <span class="sc">-</span>bins)</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>setaria_long<span class="sc">$</span>bin.labels <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"start"</span>, <span class="st">"start codon"</span>, setaria_long<span class="sc">$</span>bin.labels)</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>setaria_long<span class="sc">$</span>bin.labels <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"end"</span>, <span class="st">"stop codon"</span>, setaria_long<span class="sc">$</span>bin.labels)</span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>sorghum <span class="ot">&lt;-</span> <span class="fu">read_deeptools_table</span>(<span class="st">'~/Downloads/sorghum_meta.txt'</span>)</span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>sorghum_long <span class="ot">&lt;-</span> <span class="fu">gather</span>(sorghum, <span class="st">'sample'</span>, <span class="st">'score'</span>, <span class="sc">-</span>bin.labels, <span class="sc">-</span>bins)</span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a>sorghum_long<span class="sc">$</span>bin.labels <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"start"</span>, <span class="st">"start codon"</span>, sorghum_long<span class="sc">$</span>bin.labels)</span>
<span id="cb66-22"><a href="#cb66-22" aria-hidden="true" tabindex="-1"></a>sorghum_long<span class="sc">$</span>bin.labels <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"end"</span>, <span class="st">"stop codon"</span>, sorghum_long<span class="sc">$</span>bin.labels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And generate a combine plot to make:</p>
</section>
<section id="figure-3a" class="level1">
<h1>Figure 3A:</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make a combined plot</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>zm <span class="ot">&lt;-</span> maize_long <span class="sc">%&gt;%</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">maize =</span> <span class="dv">4</span>)</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>sb <span class="ot">&lt;-</span> sorghum_long <span class="sc">%&gt;%</span></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sorghum =</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">4</span>)</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>si <span class="ot">&lt;-</span> setaria_long <span class="sc">%&gt;%</span></span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">setaria =</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">4</span>)</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> <span class="fu">cbind</span>(zm,sb,si)</span>
<span id="cb67-15"><a href="#cb67-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-16"><a href="#cb67-16" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> combined <span class="sc">%&gt;%</span></span>
<span id="cb67-17"><a href="#cb67-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>sample) <span class="sc">%&gt;%</span></span>
<span id="cb67-18"><a href="#cb67-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(bin.labels, bins)) </span>
<span id="cb67-19"><a href="#cb67-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-20"><a href="#cb67-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(combined, <span class="fu">aes</span>(<span class="at">x =</span> bins, <span class="at">y =</span> <span class="fu">as.numeric</span>(value), <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb67-21"><a href="#cb67-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">'loess'</span>,</span>
<span id="cb67-22"><a href="#cb67-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">span =</span> .<span class="dv">2</span>,</span>
<span id="cb67-23"><a href="#cb67-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">linewidth =</span> <span class="fl">2.8</span>) <span class="sc">+</span></span>
<span id="cb67-24"><a href="#cb67-24" aria-hidden="true" tabindex="-1"></a>   ggsci<span class="sc">::</span><span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb67-25"><a href="#cb67-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> combined<span class="sc">$</span>bins,</span>
<span id="cb67-26"><a href="#cb67-26" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> combined<span class="sc">$</span>bin.labels) <span class="sc">+</span></span>
<span id="cb67-27"><a href="#cb67-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_vline(xintercept = c(start_bin, end_bin), linetype = 'dotted') +</span></span>
<span id="cb67-28"><a href="#cb67-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">'Modification density'</span>) <span class="sc">+</span></span>
<span id="cb67-29"><a href="#cb67-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">'Position'</span>) <span class="sc">+</span></span>
<span id="cb67-30"><a href="#cb67-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb67-31"><a href="#cb67-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb67-32"><a href="#cb67-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">'bottom'</span>,</span>
<span id="cb67-33"><a href="#cb67-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb67-34"><a href="#cb67-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.direction =</span> <span class="st">'horizontal'</span>,</span>
<span id="cb67-35"><a href="#cb67-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),</span>
<span id="cb67-36"><a href="#cb67-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.box.margin =</span> <span class="fu">margin</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="sc">-</span><span class="dv">10</span>),</span>
<span id="cb67-37"><a href="#cb67-37" aria-hidden="true" tabindex="-1"></a>      <span class="co">#  axis.text = element_text(color = 'black'),</span></span>
<span id="cb67-38"><a href="#cb67-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb67-39"><a href="#cb67-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb67-40"><a href="#cb67-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb67-41"><a href="#cb67-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="dv">50</span>, <span class="dv">150</span>), <span class="at">linetype =</span> <span class="st">'dotted'</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure3_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>On to the next bit.</p>
</section>
<section id="figure-3b" class="level1">
<h1>Figure 3B</h1>
<p>Exon length of modified exons vs not modified exons:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gather all expressed exons</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>expressed_tx_ids <span class="ot">&lt;-</span> tx_expressed <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(tx)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>expressed_tx_exons <span class="ot">&lt;-</span> exons <span class="sc">%&gt;%</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., expressed_tx_ids, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"group_name"</span> <span class="ot">=</span> <span class="st">"tx"</span>))</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>expressed_tx_exons<span class="sc">$</span>seqnames <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"super_"</span>, <span class="st">"super-"</span>, expressed_tx_exons<span class="sc">$</span>seqnames)</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>expressed_tx_exons <span class="ot">&lt;-</span> expressed_tx_exons <span class="sc">%&gt;%</span></span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"full_position"</span>, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep =</span> <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a><span class="co"># now gather all modified exons</span></span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a><span class="co"># This join comes from the analyses for Figure 3E</span></span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a>modified_exons <span class="ot">&lt;-</span> join <span class="sc">%&gt;%</span></span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">"full_position"</span>, exon_chr<span class="sc">:</span>exon_end, <span class="at">sep =</span> <span class="st">"_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(full_position, exon_length) <span class="sc">%&gt;%</span></span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">"modified"</span>)</span>
<span id="cb69-22"><a href="#cb69-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-23"><a href="#cb69-23" aria-hidden="true" tabindex="-1"></a>not_modified_exons <span class="ot">&lt;-</span> <span class="fu">anti_join</span>(expressed_tx_exons,</span>
<span id="cb69-24"><a href="#cb69-24" aria-hidden="true" tabindex="-1"></a>                                modified_exons, <span class="at">by =</span> <span class="st">"full_position"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-25"><a href="#cb69-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(full_position, width) <span class="sc">%&gt;%</span></span>
<span id="cb69-26"><a href="#cb69-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">exon_length =</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb69-27"><a href="#cb69-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type =</span> <span class="st">"not_modified"</span>) </span>
<span id="cb69-28"><a href="#cb69-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-29"><a href="#cb69-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-30"><a href="#cb69-30" aria-hidden="true" tabindex="-1"></a>mod_vs_not <span class="ot">&lt;-</span> <span class="fu">rbind</span>(not_modified_exons,</span>
<span id="cb69-31"><a href="#cb69-31" aria-hidden="true" tabindex="-1"></a>               modified_exons)</span>
<span id="cb69-32"><a href="#cb69-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-33"><a href="#cb69-33" aria-hidden="true" tabindex="-1"></a>mod_vs_not <span class="sc">%&gt;%</span></span>
<span id="cb69-34"><a href="#cb69-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(type) <span class="sc">%&gt;%</span></span>
<span id="cb69-35"><a href="#cb69-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">med =</span> <span class="fu">median</span>(exon_length),</span>
<span id="cb69-36"><a href="#cb69-36" aria-hidden="true" tabindex="-1"></a>            <span class="at">n =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 3
  type           med      n
  &lt;chr&gt;        &lt;int&gt;  &lt;int&gt;
1 modified       361   2677
2 not_modified   147 130433</code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(mod_vs_not<span class="sc">$</span>exon_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 148</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>mod_vs_not <span class="sc">%&gt;%</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> exon_length, <span class="at">fill =</span> type)) <span class="sc">+</span></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">linewidth =</span> <span class="fl">1.5</span>, <span class="at">alpha =</span> .<span class="dv">75</span>, <span class="at">color =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_color_npg</span>(<span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"modified"</span>,</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"not modified"</span>), <span class="at">name =</span> <span class="st">"class"</span>)<span class="sc">+</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_fill_manual(values = c("blue", "red"), labels =</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                     c("modified", "not modified")) +</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>    <span class="co">#scale_color_manual(values = c("blue", "red"), guide = "none") +</span></span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a> <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10000</span>)) <span class="sc">+</span></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a> <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Exon length (nt)"</span>) <span class="sc">+</span></span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="figure3_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Onto the last figure of Figure 3, 3F</p>
<p>This one assesses the number of isoforms expressed from each gene in the Sorghum McCormick et al tissue expression atlas:</p>
<p>This code is from the figure 2 script to load in the McCormick et al mods</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/45866739"</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="at">destfile =</span> <span class="st">"~/Downloads/sorghum_tissue_expression_atlas.txt.gz"</span>)</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sorghum_tissue_expression_atlas.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() </span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="co"># and load in this expression atlas TPM and isoforms TPM files</span></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure2_files/sorghum_expression_atlas_tpm.txt.gz?raw=TRUE"</span></span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="at">destfile =</span> <span class="st">"~/Downloads/sorghum_tissue_expression_atlas_tpm.txt.gz"</span>)</span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>tpm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/sorghum_tissue_expression_atlas_tpm.txt.gz"</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a><span class="co"># transcript TPM</span></span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure2_files/sorghum_expression_atlas_transcript_tpm.txt.gz?raw=TRUE"</span></span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="at">destfile =</span> <span class="st">"~/Downloads/sorghum_tissue_expression_atlas_tx_tpm.txt.gz"</span>)</span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a>tx_tpm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"~/Downloads/sorghum_tissue_expression_atlas_tx_tpm.txt.gz"</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>clean this mod data frame up:</p>
<p>This includes:</p>
<p>Only keeping modifications present in at least 3 replicates and</p>
<p>Removing lncRNA mods that are present on the antisense strand of a protein-coding gene. These mods are difficult to infer strand for. There are intergenic lncRNAs as well. These we will keep.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>three_reps <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(full_position, file) <span class="sc">%&gt;%</span></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(full_position) <span class="sc">%&gt;%</span></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position)</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., three_reps, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a>lnc_mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb75-17"><a href="#cb75-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">==</span> <span class="st">"lincRNA"</span> <span class="sc">|</span></span>
<span id="cb75-18"><a href="#cb75-18" aria-hidden="true" tabindex="-1"></a>           type <span class="sc">==</span> <span class="st">"antisense_lncRNA"</span>)</span>
<span id="cb75-19"><a href="#cb75-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-20"><a href="#cb75-20" aria-hidden="true" tabindex="-1"></a>native_mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb75-21"><a href="#cb75-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(type <span class="sc">!=</span> <span class="st">"lincRNA"</span> <span class="sc">&amp;</span></span>
<span id="cb75-22"><a href="#cb75-22" aria-hidden="true" tabindex="-1"></a>           type <span class="sc">!=</span> <span class="st">"antisense_lncRNA"</span>)</span>
<span id="cb75-23"><a href="#cb75-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-24"><a href="#cb75-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-25"><a href="#cb75-25" aria-hidden="true" tabindex="-1"></a><span class="co"># These next 2 steps remove modified sites that are shared between the 2 datasets. It's more likely the mods are occurring on mRNAs and not antisense lncRNAs due to expression differences</span></span>
<span id="cb75-26"><a href="#cb75-26" aria-hidden="true" tabindex="-1"></a>native_mod_sites <span class="ot">&lt;-</span> native_mods <span class="sc">%&gt;%</span></span>
<span id="cb75-27"><a href="#cb75-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position)</span>
<span id="cb75-28"><a href="#cb75-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-29"><a href="#cb75-29" aria-hidden="true" tabindex="-1"></a>lnc_mods <span class="ot">&lt;-</span> lnc_mods <span class="sc">%&gt;%</span></span>
<span id="cb75-30"><a href="#cb75-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">anti_join</span>(., native_mod_sites, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb75-31"><a href="#cb75-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-32"><a href="#cb75-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-33"><a href="#cb75-33" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">rbind</span>(native_mods, lnc_mods)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Ok, the next thing to do is establish the number of isoforms expressed per gene ID, and only analyze the “control” conditions in the McCormick expression atlas</p>
<p>They also performed stress experiments, we’ll remove those for this analysis</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># I don't think this step is needed but it's a genomicFeatures command to get transcript information for each gene and isoform</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>sb_summary <span class="ot">&lt;-</span> <span class="fu">transcriptLengths</span>(txdb, <span class="at">with.cds_len=</span>T,</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">with.utr5_len=</span>T, <span class="at">with.utr3_len=</span>T)</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>sb_gene_sum <span class="ot">&lt;-</span> sb_summary <span class="sc">%&gt;%</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_exon =</span> <span class="fu">mean</span>(nexon),</span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">tx_length =</span> <span class="fu">mean</span>(tx_len),</span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">cds_length =</span> <span class="fu">mean</span>(cds_len),</span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">utr5_length =</span> <span class="fu">mean</span>(utr5_len),</span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">utr3_length =</span> <span class="fu">mean</span>(utr3_len))</span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a><span class="co"># try to add the following information to the mods DF</span></span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a><span class="co"># all the summary information in the gene sum DF above (easy)</span></span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a><span class="co"># how many isoforms are expressed for each SRA</span></span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>tx_tpm<span class="sc">$</span>tx <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"transcript:"</span>, <span class="st">""</span>, tx_tpm<span class="sc">$</span>tx)</span>
<span id="cb76-20"><a href="#cb76-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-21"><a href="#cb76-21" aria-hidden="true" tabindex="-1"></a><span class="co"># this group_by/summarise counts the number of isoforms with at least .1 TPM and sums that up across all the input experiments to the expression atlas</span></span>
<span id="cb76-22"><a href="#cb76-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-23"><a href="#cb76-23" aria-hidden="true" tabindex="-1"></a><span class="co"># the sterss experiments have not been removed yet</span></span>
<span id="cb76-24"><a href="#cb76-24" aria-hidden="true" tabindex="-1"></a>gene_n_isoforms_expressed <span class="ot">&lt;-</span> tx_tpm <span class="sc">%&gt;%</span></span>
<span id="cb76-25"><a href="#cb76-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>tx) <span class="sc">%&gt;%</span></span>
<span id="cb76-26"><a href="#cb76-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., tx2gene, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"tx"</span> <span class="ot">=</span> <span class="st">"TXNAME"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb76-27"><a href="#cb76-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENEID, name) <span class="sc">%&gt;%</span></span>
<span id="cb76-28"><a href="#cb76-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_expressed_isoforms =</span> <span class="fu">sum</span>(value <span class="sc">&gt;</span> .<span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb76-29"><a href="#cb76-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'GENEID'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., gene_n_isoforms_expressed, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"GENEID"</span>,</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">"file"</span> <span class="ot">=</span> <span class="st">"name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., sb_gene_sum, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"gene_id"</span>))</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a><span class="co"># load in the rna-seq metadata that has the broader tissue</span></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>link <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure2_files/mccormick_rnaseq_metadata.xlsx?raw=TRUE"</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(link, <span class="st">"~/Downloads/sorghum_atlas_metadata.xlsx"</span>)</span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">"~/Downloads/sorghum_atlas_metadata.xlsx"</span>)</span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">9</span>) <span class="sc">%&gt;%</span></span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">file =</span> <span class="dv">1</span>, <span class="at">stage =</span> <span class="dv">2</span>)</span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(., metadata, <span class="at">by =</span> <span class="st">"file"</span>)</span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a><span class="co"># split out stress mods vs tissue mods</span></span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a><span class="co"># the "control" or "stress" is separated from the tissue by an underscore</span></span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a>stress_mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb78-24"><a href="#cb78-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"_"</span>, tissue))</span>
<span id="cb78-25"><a href="#cb78-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-26"><a href="#cb78-26" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb78-27"><a href="#cb78-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"_"</span>, tissue))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This next block will generate the input data for plotting. Essentially establishing a variable for modded or not, and number of isoforms expressed in each SRA</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># let's look at number of isoforms expressed vs modified or not</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="co"># x-axis number of isoforms expressed</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="co"># y-axis number % of group that is modified</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>modified_genes <span class="ot">&lt;-</span> mods <span class="sc">%&gt;%</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, file, type) <span class="sc">%&gt;%</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">modified =</span> <span class="st">"yes"</span>)</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>tissue_exps <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"_"</span>, tissue)) <span class="sc">%&gt;%</span></span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(file)</span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dtplyr)</span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>tx_tpm_dt <span class="ot">&lt;-</span> <span class="fu">lazy_dt</span>(tx_tpm)</span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a><span class="co"># isoform_modified &lt;- tx_tpm_dt %&gt;%</span></span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   pivot_longer(!tx) %&gt;%</span></span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(tx) %&gt;%</span></span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(max = max(value)) %&gt;%</span></span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   ungroup() %&gt;%</span></span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(max &gt;= .1) %&gt;%</span></span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::select(-max) %&gt;%</span></span>
<span id="cb79-25"><a href="#cb79-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   inner_join(., tissue_exps, by = c("name" = "file")) %&gt;%</span></span>
<span id="cb79-26"><a href="#cb79-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   left_join(., tx2gene, by = c("tx" = "TXNAME")) %&gt;%</span></span>
<span id="cb79-27"><a href="#cb79-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   left_join(., modified_genes, by = c("GENEID" = "gene",</span></span>
<span id="cb79-28"><a href="#cb79-28" aria-hidden="true" tabindex="-1"></a><span class="co">#                                       "name" = "file")) %&gt;%</span></span>
<span id="cb79-29"><a href="#cb79-29" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(is_modified = case_when(is.na(modified) ~ "no",</span></span>
<span id="cb79-30"><a href="#cb79-30" aria-hidden="true" tabindex="-1"></a><span class="co">#                                  !is.na(modified) ~ "yes")) %&gt;%</span></span>
<span id="cb79-31"><a href="#cb79-31" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(GENEID, name) %&gt;%</span></span>
<span id="cb79-32"><a href="#cb79-32" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(n_expressed_isoforms = sum(tpm &gt;= .1)) %&gt;%</span></span>
<span id="cb79-33"><a href="#cb79-33" aria-hidden="true" tabindex="-1"></a><span class="co">#   ungroup() %&gt;%</span></span>
<span id="cb79-34"><a href="#cb79-34" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(n_expressed_isoforms != 0) %&gt;%</span></span>
<span id="cb79-35"><a href="#cb79-35" aria-hidden="true" tabindex="-1"></a><span class="co">#   as_tibble()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-3c" class="level1">
<h1>Figure 3C</h1>
<p>A lot of this is going to come from Figure 1 to download all the HAMR/ModTect files and identify common sites</p>
<p>That will be the first code block, the second will do the isoform analysis:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># establish URLs</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>sb_h1 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SB1_hamr_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>sb_h2 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SB2_hamr_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>sb_m1 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SB1_modtect_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>sb_m2 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SB2_modtect_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Download the gzipped file</span></span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(sb_h1, <span class="at">destfile =</span> <span class="st">"~/Downloads/sbh1.txt.gz"</span>)</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(sb_h2, <span class="at">destfile =</span> <span class="st">"~/Downloads/sbh2.txt.gz"</span>)</span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(sb_m1, <span class="at">destfile =</span> <span class="st">"~/Downloads/sbm1.txt.gz"</span>)</span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(sb_m2, <span class="at">destfile =</span> <span class="st">"~/Downloads/sbm2.txt.gz"</span>)</span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the gzipped file into R</span></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>sb_h1 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sbh1.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v18<span class="sc">:</span>v20)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v21) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"sb_h1"</span>)</span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a>sb_h2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sbh2.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v18<span class="sc">:</span>v20)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v21) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"sb_h2"</span>)</span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a>sb_m1 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sbm1.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v34<span class="sc">:</span>v36)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v37) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"sb_m1"</span>)</span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a>sb_m2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sbm2.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v34<span class="sc">:</span>v36)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v37) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"sb_m2"</span>)</span>
<span id="cb80-21"><a href="#cb80-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-22"><a href="#cb80-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-23"><a href="#cb80-23" aria-hidden="true" tabindex="-1"></a>zm_h1 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/ZM1_hamr_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-24"><a href="#cb80-24" aria-hidden="true" tabindex="-1"></a>zm_h2 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/ZM2_hamr_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-25"><a href="#cb80-25" aria-hidden="true" tabindex="-1"></a>zm_m1 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/ZM1_modtect_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-26"><a href="#cb80-26" aria-hidden="true" tabindex="-1"></a>zm_m2 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/ZM2_modtect_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-27"><a href="#cb80-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-28"><a href="#cb80-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Download the gzipped files</span></span>
<span id="cb80-29"><a href="#cb80-29" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(zm_h1, <span class="at">destfile =</span> <span class="st">"~/Downloads/zmh1.txt.gz"</span>)</span>
<span id="cb80-30"><a href="#cb80-30" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(zm_h2, <span class="at">destfile =</span> <span class="st">"~/Downloads/zmh2.txt.gz"</span>)</span>
<span id="cb80-31"><a href="#cb80-31" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(zm_m1, <span class="at">destfile =</span> <span class="st">"~/Downloads/zmm1.txt.gz"</span>)</span>
<span id="cb80-32"><a href="#cb80-32" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(zm_m2, <span class="at">destfile =</span> <span class="st">"~/Downloads/zmm2.txt.gz"</span>)</span>
<span id="cb80-33"><a href="#cb80-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-34"><a href="#cb80-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Use fread from data.table to read in</span></span>
<span id="cb80-35"><a href="#cb80-35" aria-hidden="true" tabindex="-1"></a><span class="co"># do some  cleaning</span></span>
<span id="cb80-36"><a href="#cb80-36" aria-hidden="true" tabindex="-1"></a>zm_h1 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/zmh1.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v18<span class="sc">:</span>v20)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v21) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"zm_h1"</span>)</span>
<span id="cb80-37"><a href="#cb80-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-38"><a href="#cb80-38" aria-hidden="true" tabindex="-1"></a>zm_h2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/zmh2.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v18<span class="sc">:</span>v20)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v21) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"zm_h2"</span>)</span>
<span id="cb80-39"><a href="#cb80-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-40"><a href="#cb80-40" aria-hidden="true" tabindex="-1"></a>zm_m1 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/zmm1.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v34<span class="sc">:</span>v36)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v37) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"zm_m1"</span>)</span>
<span id="cb80-41"><a href="#cb80-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-42"><a href="#cb80-42" aria-hidden="true" tabindex="-1"></a>zm_m2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/zmm2.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v34<span class="sc">:</span>v36)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v37) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"zm_m2"</span>)</span>
<span id="cb80-43"><a href="#cb80-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-44"><a href="#cb80-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-45"><a href="#cb80-45" aria-hidden="true" tabindex="-1"></a>si_h1 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SI1_hamr_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-46"><a href="#cb80-46" aria-hidden="true" tabindex="-1"></a>si_h2 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SI2_hamr_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-47"><a href="#cb80-47" aria-hidden="true" tabindex="-1"></a>si_m1 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SI1_modtect_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-48"><a href="#cb80-48" aria-hidden="true" tabindex="-1"></a>si_m2 <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/SI2_modtect_clean.txt.gz?raw=TRUE"</span></span>
<span id="cb80-49"><a href="#cb80-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-50"><a href="#cb80-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Download the gzipped files</span></span>
<span id="cb80-51"><a href="#cb80-51" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(si_h1, <span class="at">destfile =</span> <span class="st">"~/Downloads/sih1.txt.gz"</span>)</span>
<span id="cb80-52"><a href="#cb80-52" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(si_h2, <span class="at">destfile =</span> <span class="st">"~/Downloads/sih2.txt.gz"</span>)</span>
<span id="cb80-53"><a href="#cb80-53" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(si_m1, <span class="at">destfile =</span> <span class="st">"~/Downloads/sim1.txt.gz"</span>)</span>
<span id="cb80-54"><a href="#cb80-54" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(si_m2, <span class="at">destfile =</span> <span class="st">"~/Downloads/sim2.txt.gz"</span>)</span>
<span id="cb80-55"><a href="#cb80-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-56"><a href="#cb80-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Use fread from data.table to read in</span></span>
<span id="cb80-57"><a href="#cb80-57" aria-hidden="true" tabindex="-1"></a><span class="co"># do some  cleaning</span></span>
<span id="cb80-58"><a href="#cb80-58" aria-hidden="true" tabindex="-1"></a>si_h1 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sih1.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v18<span class="sc">:</span>v20)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v21) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"si_h1"</span>)</span>
<span id="cb80-59"><a href="#cb80-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-60"><a href="#cb80-60" aria-hidden="true" tabindex="-1"></a>si_h2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sih2.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v18<span class="sc">:</span>v20)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v21) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"si_h2"</span>)</span>
<span id="cb80-61"><a href="#cb80-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-62"><a href="#cb80-62" aria-hidden="true" tabindex="-1"></a>si_m1 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sim1.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v34<span class="sc">:</span>v36)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v37) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"si_m1"</span>)</span>
<span id="cb80-63"><a href="#cb80-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-64"><a href="#cb80-64" aria-hidden="true" tabindex="-1"></a>si_m2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"~/Downloads/sim2.txt.gz"</span>, <span class="at">fill =</span> T, <span class="at">na.strings =</span> <span class="fu">c</span>(<span class="st">""</span>,<span class="cn">NA</span>), <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(v34<span class="sc">:</span>v36)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> v37) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">experiment =</span> <span class="st">"si_m2"</span>)</span>
<span id="cb80-65"><a href="#cb80-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-66"><a href="#cb80-66" aria-hidden="true" tabindex="-1"></a><span class="co"># make a new column to specify the site location in the chromosome</span></span>
<span id="cb80-67"><a href="#cb80-67" aria-hidden="true" tabindex="-1"></a>sb_h1 <span class="ot">&lt;-</span> sb_h1 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-68"><a href="#cb80-68" aria-hidden="true" tabindex="-1"></a>sb_h2 <span class="ot">&lt;-</span> sb_h2 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-69"><a href="#cb80-69" aria-hidden="true" tabindex="-1"></a>sb_m1 <span class="ot">&lt;-</span> sb_m1 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-70"><a href="#cb80-70" aria-hidden="true" tabindex="-1"></a>sb_m2 <span class="ot">&lt;-</span> sb_m2 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-71"><a href="#cb80-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-72"><a href="#cb80-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-73"><a href="#cb80-73" aria-hidden="true" tabindex="-1"></a><span class="co"># this bit finds the shared elements between all those unique modified sites</span></span>
<span id="cb80-74"><a href="#cb80-74" aria-hidden="true" tabindex="-1"></a>common_elements <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">c</span>(sb_h1[, <span class="dv">1</span>], sb_h2[, <span class="dv">1</span>], sb_m1[, <span class="dv">1</span>], sb_m2[,<span class="dv">1</span>])) <span class="sc">%&gt;%</span></span>
<span id="cb80-75"><a href="#cb80-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-76"><a href="#cb80-76" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">full_position =</span> <span class="dv">1</span>)</span>
<span id="cb80-77"><a href="#cb80-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-78"><a href="#cb80-78" aria-hidden="true" tabindex="-1"></a><span class="co"># only retain the common sites in these dataframes</span></span>
<span id="cb80-79"><a href="#cb80-79" aria-hidden="true" tabindex="-1"></a>sb_h1_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, sb_h1, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-80"><a href="#cb80-80" aria-hidden="true" tabindex="-1"></a>sb_h2_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, sb_h2, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-81"><a href="#cb80-81" aria-hidden="true" tabindex="-1"></a>sb_m1_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, sb_m1, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-82"><a href="#cb80-82" aria-hidden="true" tabindex="-1"></a>sb_m2_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, sb_m2, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-83"><a href="#cb80-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-84"><a href="#cb80-84" aria-hidden="true" tabindex="-1"></a>common_sorghum_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(sb_h1_common<span class="sc">$</span>gene) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-85"><a href="#cb80-85" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> <span class="dv">1</span>)</span>
<span id="cb80-86"><a href="#cb80-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-87"><a href="#cb80-87" aria-hidden="true" tabindex="-1"></a><span class="co"># maize</span></span>
<span id="cb80-88"><a href="#cb80-88" aria-hidden="true" tabindex="-1"></a>zm_h1 <span class="ot">&lt;-</span> zm_h1 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-89"><a href="#cb80-89" aria-hidden="true" tabindex="-1"></a>zm_h2 <span class="ot">&lt;-</span> zm_h2 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-90"><a href="#cb80-90" aria-hidden="true" tabindex="-1"></a>zm_m1 <span class="ot">&lt;-</span> zm_m1 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-91"><a href="#cb80-91" aria-hidden="true" tabindex="-1"></a>zm_m2 <span class="ot">&lt;-</span> zm_m2 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-92"><a href="#cb80-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-93"><a href="#cb80-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-94"><a href="#cb80-94" aria-hidden="true" tabindex="-1"></a>common_elements <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">c</span>(zm_h1[, <span class="dv">1</span>], zm_h2[, <span class="dv">1</span>], zm_m1[, <span class="dv">1</span>], zm_m2[,<span class="dv">1</span>])) <span class="sc">%&gt;%</span></span>
<span id="cb80-95"><a href="#cb80-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-96"><a href="#cb80-96" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">full_position =</span> <span class="dv">1</span>)</span>
<span id="cb80-97"><a href="#cb80-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-98"><a href="#cb80-98" aria-hidden="true" tabindex="-1"></a>zm_h1_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, zm_h1, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-99"><a href="#cb80-99" aria-hidden="true" tabindex="-1"></a>zm_h2_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, zm_h2, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-100"><a href="#cb80-100" aria-hidden="true" tabindex="-1"></a>zm_m1_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, zm_m1, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-101"><a href="#cb80-101" aria-hidden="true" tabindex="-1"></a>zm_m2_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, zm_m2, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-102"><a href="#cb80-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-103"><a href="#cb80-103" aria-hidden="true" tabindex="-1"></a><span class="co"># what are the commonly modified genes?</span></span>
<span id="cb80-104"><a href="#cb80-104" aria-hidden="true" tabindex="-1"></a>common_maize_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(zm_h1_common<span class="sc">$</span>gene) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-105"><a href="#cb80-105" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> <span class="dv">1</span>)</span>
<span id="cb80-106"><a href="#cb80-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-107"><a href="#cb80-107" aria-hidden="true" tabindex="-1"></a><span class="co"># setaria</span></span>
<span id="cb80-108"><a href="#cb80-108" aria-hidden="true" tabindex="-1"></a>si_h1 <span class="ot">&lt;-</span> si_h1 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-109"><a href="#cb80-109" aria-hidden="true" tabindex="-1"></a>si_h2 <span class="ot">&lt;-</span> si_h2 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-110"><a href="#cb80-110" aria-hidden="true" tabindex="-1"></a>si_m1 <span class="ot">&lt;-</span> si_m1 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-111"><a href="#cb80-111" aria-hidden="true" tabindex="-1"></a>si_m2 <span class="ot">&lt;-</span> si_m2 <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="at">col =</span> full_position, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="at">remove =</span> F)</span>
<span id="cb80-112"><a href="#cb80-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-113"><a href="#cb80-113" aria-hidden="true" tabindex="-1"></a>common_elements <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">c</span>(si_h1[, <span class="dv">1</span>], si_h2[, <span class="dv">1</span>], si_m1[, <span class="dv">1</span>], si_m2[,<span class="dv">1</span>])) <span class="sc">%&gt;%</span></span>
<span id="cb80-114"><a href="#cb80-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-115"><a href="#cb80-115" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">full_position =</span> <span class="dv">1</span>)</span>
<span id="cb80-116"><a href="#cb80-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-117"><a href="#cb80-117" aria-hidden="true" tabindex="-1"></a>si_h1_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, si_h1, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-118"><a href="#cb80-118" aria-hidden="true" tabindex="-1"></a>si_h2_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, si_h2, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-119"><a href="#cb80-119" aria-hidden="true" tabindex="-1"></a>si_m1_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, si_m1, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-120"><a href="#cb80-120" aria-hidden="true" tabindex="-1"></a>si_m2_common <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(common_elements, si_m2, <span class="at">by =</span> <span class="st">"full_position"</span>)</span>
<span id="cb80-121"><a href="#cb80-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-122"><a href="#cb80-122" aria-hidden="true" tabindex="-1"></a>common_setaria_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(si_h1_common<span class="sc">$</span>gene) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-123"><a href="#cb80-123" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> <span class="dv">1</span>)</span>
<span id="cb80-124"><a href="#cb80-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-125"><a href="#cb80-125" aria-hidden="true" tabindex="-1"></a><span class="co"># load in genome annotations:</span></span>
<span id="cb80-126"><a href="#cb80-126" aria-hidden="true" tabindex="-1"></a>sb_txdb <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/48356257"</span></span>
<span id="cb80-127"><a href="#cb80-127" aria-hidden="true" tabindex="-1"></a>si_txdb <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/48356254"</span></span>
<span id="cb80-128"><a href="#cb80-128" aria-hidden="true" tabindex="-1"></a>zm_txdb <span class="ot">&lt;-</span> <span class="st">"https://figshare.com/ndownloader/files/48356260"</span></span>
<span id="cb80-129"><a href="#cb80-129" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(sb_txdb, <span class="at">destfile =</span> <span class="st">"~/Downloads/sb_txdb.gff3"</span>)</span>
<span id="cb80-130"><a href="#cb80-130" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(si_txdb, <span class="at">destfile =</span> <span class="st">"~/Downloads/si_txdb.gff3"</span>)</span>
<span id="cb80-131"><a href="#cb80-131" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(zm_txdb, <span class="at">destfile =</span> <span class="st">"~/Downloads/zm_txdb.gff3"</span>)</span>
<span id="cb80-132"><a href="#cb80-132" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicFeatures)</span>
<span id="cb80-133"><a href="#cb80-133" aria-hidden="true" tabindex="-1"></a>sb_txdb <span class="ot">&lt;-</span> <span class="fu">makeTxDbFromGFF</span>(<span class="st">"~/Downloads/sb_txdb.gff3"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in call_fun_in_txdbmaker("makeTxDbFromGFF", ...): makeTxDbFromGFF() has moved to the txdbmaker package. Please call
  txdbmaker::makeTxDbFromGFF() to get rid of this warning.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Import genomic features from the file as a GRanges object ... OK
Prepare the 'metadata' data frame ... OK
Make the TxDb object ... </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .extract_exons_from_GRanges(exon_IDX, gr, mcols0, tx_IDX, feature = "exon", : 262 exons couldn't be linked to a transcript so were dropped (showing
  only the first 6):
  seqid   start     end strand   ID               Name
1     1  562993  563102      + &lt;NA&gt; ENSRNA049997130-E1
2     1 1019419 1019521      - &lt;NA&gt; ENSRNA049997137-E1
3     1 1026702 1026839      - &lt;NA&gt; ENSRNA049469069-E1
4     1 2876710 2876888      + &lt;NA&gt; ENSRNA049997145-E1
5     1 4354010 4354105      + &lt;NA&gt; ENSRNA049997152-E1
6     1 5218433 5218532      - &lt;NA&gt; ENSRNA049997158-E1
                         Parent Parent_type
1 transcript:ENSRNA049997130-T1        &lt;NA&gt;
2 transcript:ENSRNA049997137-T1        &lt;NA&gt;
3 transcript:ENSRNA049469069-T1        &lt;NA&gt;
4 transcript:ENSRNA049997145-T1        &lt;NA&gt;
5 transcript:ENSRNA049997152-T1        &lt;NA&gt;
6 transcript:ENSRNA049997158-T1        &lt;NA&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>OK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>si_txdb <span class="ot">&lt;-</span> <span class="fu">makeTxDbFromGFF</span>(<span class="st">"~/Downloads/si_txdb.gff3"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in call_fun_in_txdbmaker("makeTxDbFromGFF", ...): makeTxDbFromGFF() has moved to the txdbmaker package. Please call
  txdbmaker::makeTxDbFromGFF() to get rid of this warning.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Import genomic features from the file as a GRanges object ... OK
Prepare the 'metadata' data frame ... OK
Make the TxDb object ... </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .extract_exons_from_GRanges(exon_IDX, gr, mcols0, tx_IDX, feature = "exon", : 56 exons couldn't be linked to a transcript so were dropped (showing
  only the first 6):
  seqid    start      end strand   ID               Name
1     I  5452819  5452921      + &lt;NA&gt; ENSRNA050012413-E1
2     I  6337076  6337245      - &lt;NA&gt; ENSRNA050012400-E1
3     I  6979236  6979323      + &lt;NA&gt; ENSRNA050012398-E1
4     I 24718632 24718737      - &lt;NA&gt; ENSRNA050012494-E1
5     I 32901687 32901784      + &lt;NA&gt; ENSRNA050012302-E1
6     I 37871088 37871198      - &lt;NA&gt; ENSRNA050012319-E1
                         Parent Parent_type
1 transcript:ENSRNA050012413-T1        &lt;NA&gt;
2 transcript:ENSRNA050012400-T1        &lt;NA&gt;
3 transcript:ENSRNA050012398-T1        &lt;NA&gt;
4 transcript:ENSRNA050012494-T1        &lt;NA&gt;
5 transcript:ENSRNA050012302-T1        &lt;NA&gt;
6 transcript:ENSRNA050012319-T1        &lt;NA&gt;</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>OK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>zm_txdb <span class="ot">&lt;-</span> <span class="fu">makeTxDbFromGFF</span>(<span class="st">"~/Downloads/zm_txdb.gff3"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in call_fun_in_txdbmaker("makeTxDbFromGFF", ...): makeTxDbFromGFF() has moved to the txdbmaker package. Please call
  txdbmaker::makeTxDbFromGFF() to get rid of this warning.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Import genomic features from the file as a GRanges object ... OK
Prepare the 'metadata' data frame ... OK
Make the TxDb object ... OK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load in TPM</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>sb_tpm <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/sorghum_tpm.txt?raw=TRUE"</span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(sb_tpm, <span class="at">destfile =</span> <span class="st">"~/Downloads/sb_tpm.txt"</span>)</span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>sb_tpm <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/sb_tpm.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col_names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 34779 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (1): Gene
dbl (1): tpm_avg

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>zm_tpm <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/maize_tpm.txt?raw=TRUE"</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(zm_tpm, <span class="at">destfile =</span> <span class="st">"~/Downloads/zm_tpm.txt"</span>)</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>zm_tpm <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/zm_tpm.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col_names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 43460 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (1): Gene
dbl (1): tpm_avg

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>si_tpm <span class="ot">&lt;-</span> <span class="st">"https://github.com/kylepalos/comparative-PRM-paper/blob/main/figure1_files/setaria_tpm.txt?raw=TRUE"</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(si_tpm, <span class="at">destfile =</span> <span class="st">"~/Downloads/si_tpm.txt"</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>si_tpm <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"~/Downloads/si_tpm.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">col_names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 36241 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (1): Gene
dbl (1): tpm_avg

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>code for processing and plotting 3C</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>sb <span class="ot">&lt;-</span> <span class="fu">transcriptLengths</span>(sb_txdb) <span class="sc">%&gt;%</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">n_exon =</span> <span class="fu">mean</span>(nexon), <span class="at">mean_len =</span> <span class="fu">mean</span>(tx_len)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>zm <span class="ot">&lt;-</span> <span class="fu">transcriptLengths</span>(zm_txdb) <span class="sc">%&gt;%</span></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">n_exon =</span> <span class="fu">mean</span>(nexon), <span class="at">mean_len =</span> <span class="fu">mean</span>(tx_len)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>si <span class="ot">&lt;-</span> <span class="fu">transcriptLengths</span>(si_txdb) <span class="sc">%&gt;%</span></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">n_exon =</span> <span class="fu">mean</span>(nexon), <span class="at">mean_len =</span> <span class="fu">mean</span>(tx_len)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>sb <span class="ot">&lt;-</span> sb <span class="sc">%&gt;%</span></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon =</span> <span class="fu">case_when</span>(n_exon <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"mono-exonic"</span>,</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>                           n_exon <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"multi-exonic"</span>))</span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>zm <span class="ot">&lt;-</span> zm <span class="sc">%&gt;%</span></span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon =</span> <span class="fu">case_when</span>(n_exon <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"mono-exonic"</span>,</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>                          n_exon <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"multi-exonic"</span>))</span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a>si <span class="ot">&lt;-</span> si <span class="sc">%&gt;%</span></span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exon =</span> <span class="fu">case_when</span>(n_exon <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"mono-exonic"</span>,</span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a>                          n_exon <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"multi-exonic"</span>))</span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a>sb_mods <span class="ot">&lt;-</span> sb_h1_common <span class="sc">%&gt;%</span></span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position, gene) <span class="sc">%&gt;%</span></span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_mods =</span> <span class="fu">n_distinct</span>(full_position)) <span class="sc">%&gt;%</span></span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sb, ., <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span> <span class="ot">=</span> <span class="st">"gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_mods =</span> <span class="fu">case_when</span>(<span class="fu">is.na</span>(n_mods) <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">!</span><span class="fu">is.na</span>(n_mods) <span class="sc">~</span> n_mods))</span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a>zm_mods <span class="ot">&lt;-</span> zm_h1_common <span class="sc">%&gt;%</span></span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position, gene) <span class="sc">%&gt;%</span></span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_mods =</span> <span class="fu">n_distinct</span>(full_position)) <span class="sc">%&gt;%</span></span>
<span id="cb99-35"><a href="#cb99-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb99-36"><a href="#cb99-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(zm, ., <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span> <span class="ot">=</span> <span class="st">"gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-37"><a href="#cb99-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_mods =</span> <span class="fu">case_when</span>(<span class="fu">is.na</span>(n_mods) <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb99-38"><a href="#cb99-38" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">!</span><span class="fu">is.na</span>(n_mods) <span class="sc">~</span> n_mods))</span>
<span id="cb99-39"><a href="#cb99-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-40"><a href="#cb99-40" aria-hidden="true" tabindex="-1"></a>si_mods <span class="ot">&lt;-</span> si_h1_common <span class="sc">%&gt;%</span></span>
<span id="cb99-41"><a href="#cb99-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(full_position, gene) <span class="sc">%&gt;%</span></span>
<span id="cb99-42"><a href="#cb99-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb99-43"><a href="#cb99-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_mods =</span> <span class="fu">n_distinct</span>(full_position)) <span class="sc">%&gt;%</span></span>
<span id="cb99-44"><a href="#cb99-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb99-45"><a href="#cb99-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(si, ., <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span> <span class="ot">=</span> <span class="st">"gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-46"><a href="#cb99-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_mods =</span> <span class="fu">case_when</span>(<span class="fu">is.na</span>(n_mods) <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb99-47"><a href="#cb99-47" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">!</span><span class="fu">is.na</span>(n_mods) <span class="sc">~</span> n_mods))</span>
<span id="cb99-48"><a href="#cb99-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-49"><a href="#cb99-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-50"><a href="#cb99-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-51"><a href="#cb99-51" aria-hidden="true" tabindex="-1"></a>sb_tpm <span class="sc">%&gt;%</span></span>
<span id="cb99-52"><a href="#cb99-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(tpm_avg <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-53"><a href="#cb99-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Gene) <span class="ot">-&gt;</span> sb_tpm</span>
<span id="cb99-54"><a href="#cb99-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-55"><a href="#cb99-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-56"><a href="#cb99-56" aria-hidden="true" tabindex="-1"></a>zm_tpm <span class="ot">&lt;-</span> zm_tpm <span class="sc">%&gt;%</span></span>
<span id="cb99-57"><a href="#cb99-57" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(tpm_avg <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-58"><a href="#cb99-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Gene)</span>
<span id="cb99-59"><a href="#cb99-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-60"><a href="#cb99-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-61"><a href="#cb99-61" aria-hidden="true" tabindex="-1"></a>si_tpm <span class="ot">&lt;-</span> si_tpm <span class="sc">%&gt;%</span></span>
<span id="cb99-62"><a href="#cb99-62" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(tpm_avg <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-63"><a href="#cb99-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Gene)</span>
<span id="cb99-64"><a href="#cb99-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-65"><a href="#cb99-65" aria-hidden="true" tabindex="-1"></a>sb_mods <span class="ot">&lt;-</span> sb_mods <span class="sc">%&gt;%</span></span>
<span id="cb99-66"><a href="#cb99-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., sb_tpm, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-67"><a href="#cb99-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">modified =</span> <span class="fu">case_when</span>(n_mods <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"not_modified"</span>,</span>
<span id="cb99-68"><a href="#cb99-68" aria-hidden="true" tabindex="-1"></a>                              n_mods <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"modified"</span>),</span>
<span id="cb99-69"><a href="#cb99-69" aria-hidden="true" tabindex="-1"></a>         <span class="at">species =</span> <span class="st">"Sorghum"</span>)</span>
<span id="cb99-70"><a href="#cb99-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-71"><a href="#cb99-71" aria-hidden="true" tabindex="-1"></a>zm_mods <span class="ot">&lt;-</span> zm_mods <span class="sc">%&gt;%</span></span>
<span id="cb99-72"><a href="#cb99-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., zm_tpm, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-73"><a href="#cb99-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">modified =</span> <span class="fu">case_when</span>(n_mods <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"not_modified"</span>,</span>
<span id="cb99-74"><a href="#cb99-74" aria-hidden="true" tabindex="-1"></a>                              n_mods <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"modified"</span>),</span>
<span id="cb99-75"><a href="#cb99-75" aria-hidden="true" tabindex="-1"></a>         <span class="at">species =</span> <span class="st">"Maize"</span>)</span>
<span id="cb99-76"><a href="#cb99-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-77"><a href="#cb99-77" aria-hidden="true" tabindex="-1"></a>si_mods <span class="ot">&lt;-</span> si_mods <span class="sc">%&gt;%</span></span>
<span id="cb99-78"><a href="#cb99-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(., si_tpm, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span> <span class="ot">=</span> <span class="st">"Gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-79"><a href="#cb99-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">modified =</span> <span class="fu">case_when</span>(n_mods <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"not_modified"</span>,</span>
<span id="cb99-80"><a href="#cb99-80" aria-hidden="true" tabindex="-1"></a>                              n_mods <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"modified"</span>),</span>
<span id="cb99-81"><a href="#cb99-81" aria-hidden="true" tabindex="-1"></a>         <span class="at">species =</span> <span class="st">"Setaria"</span>)</span>
<span id="cb99-82"><a href="#cb99-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-83"><a href="#cb99-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-84"><a href="#cb99-84" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sb_mods,</span>
<span id="cb99-85"><a href="#cb99-85" aria-hidden="true" tabindex="-1"></a>                  zm_mods,</span>
<span id="cb99-86"><a href="#cb99-86" aria-hidden="true" tabindex="-1"></a>                  si_mods)</span>
<span id="cb99-87"><a href="#cb99-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-88"><a href="#cb99-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-89"><a href="#cb99-89" aria-hidden="true" tabindex="-1"></a>combined <span class="sc">%&gt;%</span></span>
<span id="cb99-90"><a href="#cb99-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(species, exon) <span class="sc">%&gt;%</span></span>
<span id="cb99-91"><a href="#cb99-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_modified =</span> <span class="fu">sum</span>(n_mods <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb99-92"><a href="#cb99-92" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_unmodified =</span> <span class="fu">sum</span>(n_mods <span class="sc">==</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb99-93"><a href="#cb99-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb99-94"><a href="#cb99-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">proportion_modified =</span> n_modified <span class="sc">/</span> (n_modified <span class="sc">+</span> n_unmodified)) <span class="sc">%&gt;%</span></span>
<span id="cb99-95"><a href="#cb99-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> species, <span class="at">y =</span> <span class="dv">100</span><span class="sc">*</span>(proportion_modified), <span class="at">fill =</span> exon)) <span class="sc">+</span></span>
<span id="cb99-96"><a href="#cb99-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position=</span><span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb99-97"><a href="#cb99-97" aria-hidden="true" tabindex="-1"></a>  ggsci<span class="sc">::</span><span class="fu">scale_fill_npg</span>(<span class="at">name =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb99-98"><a href="#cb99-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">12</span>) <span class="sc">+</span></span>
<span id="cb99-99"><a href="#cb99-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Proportion of expressed</span><span class="sc">\n</span><span class="st">genes modified"</span>) <span class="sc">+</span></span>
<span id="cb99-100"><a href="#cb99-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb99-101"><a href="#cb99-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'species'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="figure3_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>